<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Changelog • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Changelog"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a>
    <a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a>
    <a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/wejlab/MetaScope">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-news">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Changelog</h1>
      <small>Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/NEWS.md" class="external-link"><code>NEWS.md</code></a></small>
    </div>

    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.3.7" id="metascope-137-bioc-319">MetaScope 1.3.7 (Bioc 3.19)<a class="anchor" aria-label="anchor" href="#metascope-137-bioc-319"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-3-7">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-3-7"></a></h3>
<ul><li>Identified the Rbowtie2 parameter k as being doubled when specified for bowtie filter or align steps.</li>
<li>Added another call to taxize in <code><a href="../reference/convert_animalcules.html">convert_animalcules()</a></code> to catch any accessions that were not mapped to a UID in <code><a href="../reference/metascope_id.html">metascope_id()</a></code>, in addition to another call in <code><a href="../reference/metascope_id.html">metascope_id()</a></code> itself</li>
</ul></div>
<div class="section level3">
<h3 id="major-changes-1-3-7">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-3-7"></a></h3>
<ul><li>Altered <code>bt2_params</code> objects to reflect 98% identity (16S), 95% identity (metagenomics) and added a parameter for when the origin genome is thought to not be present in the reference database.</li>
<li>Added SILVA <code>species_headers</code> object called internally in <code>convert_animalcules_silva</code>
</li>
<li>Added <code>convert_animalcules_silva</code> function</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="1.2.0" id="metascope-120-bioc-318">MetaScope 1.2.0 (Bioc 3.18)<a class="anchor" aria-label="anchor" href="#metascope-120-bioc-318"></a></h2>
<div class="section level3">
<h3 id="bug-fixes-1-2-0">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-1-2-0"></a></h3>
<ul><li>Fixed taxonomy table function to output correctly formatted table</li>
<li>Fixed examples for various functions that were calling genomes with download_refseq but genomes were not able to be found.</li>
<li>Fixed plot generation for <code><a href="../reference/metascope_id.html">metascope_id()</a></code>
</li>
<li>Fixed premature stopping of <code>download_refseq</code> for strains labeled as “no rank” in NCBI.</li>
<li>Fixed identification of reads as unknown genomes (due to outdated reference databases that identify genomes now removed from NCBI). The unknown genomes will now be distinctly identified based on NCBI accession ID, which will separate them in the final results. IDs can also be looked up manually (on NCBI website) to see what the reads were aligning to.</li>
<li>Incorporated unknown taxa (removed from NCBI databases) into output of <code><a href="../reference/convert_animalcules.html">convert_animalcules()</a></code>
</li>
</ul></div>
<div class="section level3">
<h3 id="major-changes-1-2-0">Major changes<a class="anchor" aria-label="anchor" href="#major-changes-1-2-0"></a></h3>
<ul><li>Added ability to identify reads from databases other than NCBI (known to work for Silva and Greengenes2)</li>
</ul></div>
</div>
    <div class="section level2">
<h2 class="pkg-version" data-toc-text="0.99.0" id="metascope-0990-4192022">MetaScope 0.99.0 (4/19/2022)<a class="anchor" aria-label="anchor" href="#metascope-0990-4192022"></a></h2>
<ul><li>Pre-Release version of MetaScope</li>
</ul><div class="section level3">
<h3 id="bug-fixes-0-99-0">Bug Fixes<a class="anchor" aria-label="anchor" href="#bug-fixes-0-99-0"></a></h3>
<ul><li>Fixed check error message about <code><a href="https://rdrr.io/pkg/data.table/man/fread.html" class="external-link">data.table::fread</a></code> for reading .gz files by adding <code>R.utils</code> to imports.</li>
</ul></div>
<div class="section level3">
<h3 id="major-changes-0-99-0">Major Changes<a class="anchor" aria-label="anchor" href="#major-changes-0-99-0"></a></h3>
<ul><li>Submitted to Bioconductor</li>
</ul></div>
<div class="section level3">
<h3 id="minor-changes-0-99-0">Minor Changes<a class="anchor" aria-label="anchor" href="#minor-changes-0-99-0"></a></h3>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

