<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MetaScope">
<title>Introduction to MetaScope • MetaScope</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to MetaScope">
<meta property="og:description" content="MetaScope">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a>
    <a class="dropdown-item" href="../articles/MetaScopeTutorial.html">Condensed MetaScope Tutorial</a>
    <a class="dropdown-item" href="../articles/docs/SILVA.html">Usage with SILVA 16S Database</a>
    <a class="dropdown-item" href="../articles/docs/Example_Script.html">Example MetaScope Script</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/wejlab/MetaScope">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Introduction to MetaScope</h1>
                        <h4 data-toc-skip class="author">Aubrey
Odom</h4>
            <address class="author_afil">
      Program in Bioinformatics, Boston University, Boston,
MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:aodom@bu.edu" class="email">aodom@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">Rahul
Varki</h4>
            <address class="author_afil">
      Research Assistant in Johnson Lab, Boston University School of
Medicine, Boston,
MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:rvarki@bu.edu" class="email">rvarki@bu.edu</a>
      </address>
                              <h4 data-toc-skip class="author">W. Evan
Johnson</h4>
            <address class="author_afil">
      The Section of Computational Biomedicine, Boston University School
of Medicine, Boston,
MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:wej@bu.edu" class="email">wej@bu.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">October 30, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/vignettes/MetaScope_vignette.Rmd" class="external-link"><code>vignettes/MetaScope_vignette.Rmd</code></a></small>
      <div class="d-none name"><code>MetaScope_vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="important-note">Important Note<a class="anchor" aria-label="anchor" href="#important-note"></a>
</h2>
<p>Although it is not a requirement, MetaScope will fun faster and more
efficiently for larger samples if the samtools package is present on
your device. You can download it at <a href="https://github.com/samtools/samtools" class="external-link uri">https://github.com/samtools/samtools</a>.</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>MetaScope is a complete metagenomics profiling package that can
accurately identify the composition of microbes within a sample at a
strain-level resolution. MetaScope can be considered as an updated and
expanded R translation of <a href="https://microbiomejournal.biomedcentral.com/articles/10.1186/2049-2618-2-33" class="external-link">PathoScope
2.0</a>, a Python-based metagenomic profiling package created by the
Johnson lab. A few improvements made in MetaScope include using the BAM
file format instead of the SAM file format for significantly less disk
space usage, removing all dependencies to NCBI’s now defunct GI sequence
annotations, and properly filtering reads that align to filter reference
genomes. Functions to analyze host microbiome data are also planned to
be added in future updates to the package.</p>
<div class="section level3">
<h3 id="metascope-workflows">MetaScope Workflows<a class="anchor" aria-label="anchor" href="#metascope-workflows"></a>
</h3>
<p>The major workflow of MetaScope is delineated below. It is composed
of core modules that are formed by groups of functions.</p>
<div class="float">
<img src="MetaScope_Workflow.png" alt="Figure 1. The MetaScope workflow and its associated modules with function descriptions. The MetaRef, MetaAlign, MetaFilter, and MetaID modules form the backbone of package operation, whereas the MetaDemultiplex, MetaBLAST, and MetaCombine modules are complementary to the core package functionality."><div class="figcaption">Figure 1. The MetaScope workflow and its
associated modules with function descriptions. The MetaRef, MetaAlign,
MetaFilter, and MetaID modules form the backbone of package operation,
whereas the MetaDemultiplex, MetaBLAST, and MetaCombine modules are
complementary to the core package functionality.</div>
</div>
<p>The core modules are as follows: 1. <strong>MetaDemultiplex:</strong>
Obtain non-barcoded sequencing reads 2. <strong>MetaRef:</strong> Obtain
target and filter genome sequences from NCBI nucleotide database and
index using a given aligner 3. <strong>MetaAlign:</strong> Align
sequencing reads to indexed target genome sequences 4.
<strong>MetaFilter:</strong> Remove reads mapped to indexed host genome
sequences 5. <strong>MetaID:</strong> Reassign ambiguously mapped reads
to probable source genome 6. <strong>MetaBLAST:</strong> BLAST assigned
reads against the NCBI nucleotide database to check identity 7.
<strong>MetaCombine:</strong> Aggregate samples into a
MultiAssayExperiment compatible with the <code>animalcules</code> R
package.</p>
<p>There are two sub-workflows that are included in the package, as seen
in Figure 1: the Rbowtie2 and the Rsubread workflow. The major
difference is that the functions in the MetaRef, MetaAlign, and
MetaFilter modules differ by the aligner utilized.</p>
<p>The Rbowtie2 functions utilize the <a href="https://www.nature.com/articles/nmeth.1923" class="external-link">Bowtie2</a> aligner
(Langmead 2012) whereas the <a href="https://academic.oup.com/nar/article/47/8/e47/5345150" class="external-link">Rsubread</a>
functions utilize the Rsubread aligner (Liao 2019). The nuances of how
to use each function can be found by looking at each function’s help
manual (R command: <code>?&lt;name of function&gt;</code>). For
reference, PathoScope 2.0 uses the Bowtie2 aligner in its workflow.</p>
<p>In this vignette, we will analyze the mock data provided in the
package via the Rbowtie2 sub-workflow. We will utilize all of the core
modules in sequential order. We will make mention of the equivalent
Rsubread function whenever an Rbowtie2 function is being used. For the
purposes of this example, MetaBLAST and MetaCombine modules will be
omitted.</p>
</div>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<p>In order to install MetaScope from Bioconductor, run the following
code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"MetaScope"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wejlab/metascope" class="external-link">MetaScope</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://magrittr.tidyverse.org" class="external-link">magrittr</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="entrez-key">Entrez key<a class="anchor" aria-label="anchor" href="#entrez-key"></a>
</h3>
<p>We highly recommend grabbing an Entrez key using the function
<code><a href="https://docs.ropensci.org/taxize/reference/key_helpers.html" class="external-link">taxize::use_entrez()</a></code>. Then set your key as a global
environment variable like so:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NCBI_key</span> <span class="op">&lt;-</span> <span class="st">"&lt;Your key here&gt;"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span><span class="st">"ENTREZ_KEY"</span> <span class="op">=</span> <span class="va">NCBI_key</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>The mock data provided in the package consists of simulated
sequencing data generated from the <code>SAMtools</code>
<code>wgsim</code> function (see <code>extdata_explanations.Rmd</code>
in <code>inst/script</code> to see exact commands). The
<code>wgsim</code> function is a tool which allows for the generation of
FASTQ reads from a reference genome (FASTA). The mock data
(<code>reads.fastq</code>) contains 1500 reads, of which 1000 reads are
derived from the <em>Staphylococcus aureus</em> RF122 strain and 500
reads are derived from the <em>Staphylococcus epidermidis</em> RP62A
strain. In this data set, we assume that the <em>S. aureus</em> RF122
reads are the reads of interest, and the <em>S. epidermidis</em> RP62A
reads are known contaminant reads which should be removed during the
analysis. Ideally, the microbial composition report (.csv) produced at
the end of the analysis should contain only reads assigned to the <em>S.
aureus</em> RF122 strain.</p>
</div>
<div class="section level2">
<h2 id="metademultiplex-demultiplexing-reads">MetaDemultiplex: Demultiplexing reads<a class="anchor" aria-label="anchor" href="#metademultiplex-demultiplexing-reads"></a>
</h2>
<p>Sequence runs on NGS instruments are typically carried out with
multiple samples pooled together. An index tag (also called a barcode)
consisting of a unique sequence of between 6 and 12bp is added to each
sample so that the sequence reads from different samples can be
identified. For 16s experiments or sequencing conducted using an
Illumina machine, the process of demultiplexing (dividing your sequence
reads into separate files for each index tag/sample) and generating the
FASTQ data files required for downstream analysis can be done using the
MetaScope demultiplexing workflow. This consists of the
<code><a href="../reference/meta_demultiplex.html">meta_demultiplex()</a></code> function, which takes as arguments a
matrix of sample names/barcodes, a FASTQ file of barcodes by sequence
header, and a FASTQ file of reads corresponding to the barcodes. Based
on the barcodes given, the function extracts all reads for the indexed
barcode and writes all the reads from that barcode to separate FASTQ
files.</p>
<p>This is an optional step in the analysis process depending on whether
your reads are multiplexed. The reads which we are currently trying to
analyze are not multiplexed and therefore this step is skipped in our
analysis. The example shown below is using different reads that are
barcoded in order to show the utility of the function.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get barcode, index, and read data locations</span></span>
<span><span class="va">barcodePath</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"barcodes.txt"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span><span class="va">indexPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example_index.fastq"</span>,</span>
<span>                         package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span><span class="va">readPath</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example.fastq"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get barcode, index, and read data locations</span></span>
<span><span class="va">demult</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/meta_demultiplex.html">meta_demultiplex</a></span><span class="op">(</span><span class="va">barcodePath</span>,</span>
<span>                   <span class="va">indexPath</span>,</span>
<span>                   <span class="va">readPath</span>,</span>
<span>                   rcBarcodes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                   hammingDist <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                   location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Warning in XStringSet("DNA", x, start = start, end = end, width = width, :</span></span>
<span><span class="co">## metadata columns on input DNAStringSet object were dropped</span></span></code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">demult</span></span></code></pre></div>
<pre><code><span><span class="co">##   SampleName Barcode NumberOfReads</span></span>
<span><span class="co">## 1        CDV TCCACGT            25</span></span>
<span><span class="co">## 2   LaCrosse ACAGGCT            25</span></span>
<span><span class="co">## 3        RSV ATCGTGC            25</span></span>
<span><span class="co">## 4       EboV ACTACAG            25</span></span>
<span><span class="co">## 5    Measles AAGTCGC            25</span></span>
<span><span class="co">## 6        VSV TCTCAGG            25</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="metaref-reference-genome-library">MetaRef: Reference Genome Library<a class="anchor" aria-label="anchor" href="#metaref-reference-genome-library"></a>
</h2>
<p>The MetaScope genome library workflow is designed to assist with
collection of sequence data from the National Center for Biotechnology
Information (NCBI) nucleotide database. Prior to doing so, the potential
targets and filters for the analysis should be identified. That is, what
“target” species do you expect to find in your metagenomic sample that
you would like to identify, and what reads would you like to “filter”
out from the data that are not essential to your analysis?</p>
<p>Typically, the targets of the analysis are microbes (that is,
viruses, bacteria, and fungi), and we wish to filter out or discard any
reads from the host in addition to artificially added sequences, such as
Phi X 174. Following identification of the targets and filters, we use a
reference genome library to align the vast number of sample reads back
to the respective regions of origin in various species.</p>
<p>The <code><a href="../reference/download_refseq.html">download_refseq()</a></code> function automatically extracts
custom reference genome libraries in a FASTA file for microbial or host
genomes. The user must first indicate a taxon for which to download the
genomes, such as ‘bacteria’ or ‘primates’. A table of possible entries
can be viewed by accessing the <code>MetaScope:::taxonomy_table</code>
object. The user may then specify whether they wish to download only the
RefSeq reference genomes, or both the reference and representative
genomes. The compress option then allows users to specify whether to
compress the output FASTA file; this is done by default.</p>
<div class="section level3">
<h3 id="downloading-target-genomes">Downloading target genomes<a class="anchor" aria-label="anchor" href="#downloading-target-genomes"></a>
</h3>
<p>Even though in this scenario we know exactly from where the reads in
the mock data (<code>reads.fastq</code>) originate, in most cases we may
only have a general idea of read origins. Therefore, in the following
code, we will download the genome of the <em>Staphylococcus aureus</em>
RF122 strain along with the genomes of a few other closely related
<em>Staphylococcus aureus</em> strains from the NCBI RefSeq database.
These genomes together will act as our target genome library.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">target_ref_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">target_ref_temp</span><span class="op">)</span></span>
<span></span>
<span><span class="va">all_species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Staphylococcus aureus subsp. aureus Mu50"</span>,</span>
<span>                 <span class="st">"Staphylococcus aureus subsp. aureus Mu3"</span>,</span>
<span>                 <span class="st">"Staphylococcus aureus subsp. aureus str. Newman"</span>,</span>
<span>                 <span class="st">"Staphylococcus aureus subsp. aureus N315"</span>,</span>
<span>                 <span class="st">"Staphylococcus aureus RF122"</span>, <span class="st">"Staphylococcus aureus subsp. aureus ST398"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">all_species</span>, <span class="va">download_refseq</span>, </span>
<span>       reference <span class="op">=</span> <span class="cn">FALSE</span>, representative <span class="op">=</span> <span class="cn">FALSE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>       out_dir <span class="op">=</span> <span class="va">target_ref_temp</span>, caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##                                                                       Staphylococcus aureus subsp. aureus Mu50 </span></span>
<span><span class="co">##        "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc5430035f/Staphylococcus_aureus_subsp._aureus_Mu50.fasta.gz" </span></span>
<span><span class="co">##                                                                        Staphylococcus aureus subsp. aureus Mu3 </span></span>
<span><span class="co">##         "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc5430035f/Staphylococcus_aureus_subsp._aureus_Mu3.fasta.gz" </span></span>
<span><span class="co">##                                                                Staphylococcus aureus subsp. aureus str. Newman </span></span>
<span><span class="co">## "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc5430035f/Staphylococcus_aureus_subsp._aureus_str._Newman.fasta.gz" </span></span>
<span><span class="co">##                                                                       Staphylococcus aureus subsp. aureus N315 </span></span>
<span><span class="co">##        "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc5430035f/Staphylococcus_aureus_subsp._aureus_N315.fasta.gz" </span></span>
<span><span class="co">##                                                                                    Staphylococcus aureus RF122 </span></span>
<span><span class="co">##                     "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc5430035f/Staphylococcus_aureus_RF122.fasta.gz" </span></span>
<span><span class="co">##                                                                      Staphylococcus aureus subsp. aureus ST398 </span></span>
<span><span class="co">##       "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc5430035f/Staphylococcus_aureus_subsp._aureus_ST398.fasta.gz"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="downloading-filter-genomes">Downloading filter genomes<a class="anchor" aria-label="anchor" href="#downloading-filter-genomes"></a>
</h3>
<p>We will also download the reference genome and related sequences* of
the <em>Staphylococcus epidermidis</em> RP62A strain from the NCBI
nucleotide database, in an uncompressed FASTA format. This genome will
act as our filter library.</p>
<ul>
<li>since we are downloading from the nucleotide database with
representative and reference = FALSE, several sequences will be
downloaded in addition to the main genome.</li>
</ul>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">filter_ref_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">filter_ref_temp</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="../reference/download_refseq.html">download_refseq</a></span><span class="op">(</span></span>
<span>  taxon <span class="op">=</span> <span class="st">"Staphylococcus epidermidis RP62A"</span>,</span>
<span>  representative <span class="op">=</span> <span class="cn">FALSE</span>, reference <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  compress <span class="op">=</span> <span class="cn">TRUE</span>, out_dir <span class="op">=</span> <span class="va">filter_ref_temp</span>,</span>
<span>  caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc25755482/Staphylococcus_epidermidis_RP62A.fasta.gz"</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="creating-indices-using-a-given-aligner">Creating indices using a given aligner<a class="anchor" aria-label="anchor" href="#creating-indices-using-a-given-aligner"></a>
</h3>
<p>We now use <code><a href="../reference/mk_bowtie_index.html">mk_bowtie_index()</a></code>, a wrapper for the
<code><a href="https://rdrr.io/pkg/Rbowtie2/man/bowtie2-build.html" class="external-link">Rbowtie2::bowtie2_build</a></code> function, to generate Bowtie2
compatible indexes from the reference genomes that were previously
downloaded. The target and reference genome files (.fasta or .fasta.gz
extension) must be placed into their own separate empty directories
prior to using the function. This is due to the fact that the function
will attempt to build the indexes from all the files present in the
directory. The function will give an error if other files (other than
.fasta or .fasta.gz) are present in the directory. Depending on the
combined size of the reference genomes, the function will automatically
create either small (.bt2) or large (.bt2l) Bowtie2 indexes.</p>
<p>The target and filter reference genomes downloaded in the previous
step have been combined and renamed to target.fasta and filter.fasta
respectively for convenience. These are the files from which the Bowtie2
indexes will be made from.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create temp directory to store the Bowtie2 indices</span></span>
<span><span class="va">index_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">index_temp</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create target index</span></span>
<span><span class="fu"><a href="../reference/mk_bowtie_index.html">mk_bowtie_index</a></span><span class="op">(</span></span>
<span>  ref_dir <span class="op">=</span> <span class="va">target_ref_temp</span>,</span>
<span>  lib_dir <span class="op">=</span> <span class="va">index_temp</span>,</span>
<span>  lib_name <span class="op">=</span> <span class="st">"target"</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## arguments 'show.output.on.console', 'minimized' and 'invisible' are for Windows only</span></span></code></pre>
<pre><code><span><span class="co">## Index building complete</span></span></code></pre>
<pre><code><span><span class="co">## [1] "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc4fc2142d"</span></span></code></pre>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create filter index</span></span>
<span><span class="fu"><a href="../reference/mk_bowtie_index.html">mk_bowtie_index</a></span><span class="op">(</span></span>
<span>  ref_dir <span class="op">=</span> <span class="va">filter_ref_temp</span>,</span>
<span>  lib_dir <span class="op">=</span> <span class="va">index_temp</span>,</span>
<span>  lib_name <span class="op">=</span> <span class="st">"filter"</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## arguments 'show.output.on.console', 'minimized' and 'invisible' are for Windows only</span></span>
<span><span class="co">## Index building complete</span></span></code></pre>
<pre><code><span><span class="co">## [1] "/scratch/290807.1.ood/RtmpBNysHC/file3b99fc4fc2142d"</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="alignment-with-reference-libraries">Alignment with Reference Libraries<a class="anchor" aria-label="anchor" href="#alignment-with-reference-libraries"></a>
</h2>
<p>After acquiring the target and filter genome libraries, we then take
the sequencing reads from our sample and map them first to the target
library and then to the filter library. MetaScope’s Rbowtie2 mapping
function utilizes the <a href="https://www.nature.com/articles/nmeth.1923" class="external-link">Bowtie2</a> aligner
(Langmead 2012) which maps reads to a reference genome using a full-text
minute index based approach. Essentially, the algorithm extracts
substrings which are referred to as “seeds” from the reads and aligns
them to the reference genomes with the assistance from the full-text
minute index. Seed alignments to the reference genomes are prioritized
and then finally extended into full alignments using dynamic
programming.</p>
<div class="section level3">
<h3 id="metaalign">MetaAlign<a class="anchor" aria-label="anchor" href="#metaalign"></a>
</h3>
<p>Following index creation, we will use the Bowtie2 aligner to map the
reads to the target genomes with the <code><a href="../reference/align_target_bowtie.html">align_target_bowtie()</a></code>
function (Rsubread equivalent: <code><a href="../reference/align_target.html">align_target()</a></code>). The
function takes as an input the location of the FASTQ file to align, the
directory where the indexes are stored, the names of the indexes to
align against, the directory where the BAM file should be written, and
the basename of the output BAM file.</p>
<p>In practice, <code><a href="../reference/align_target_bowtie.html">align_target_bowtie()</a></code> maps reads to each
target library separately, removes the unmapped reads from each file,
and finally merges and sorts by chromosome the BAM files from each
library into a single output file (same with <code>align_target</code>).
If <code>SAMtools</code> is installed on the machine and can be found by
the <code>Sys.which("samtools")</code> R command, the BAM file will be
directly created, otherwise an intermediate SAM file will be created
prior to the creation of the BAM file which could potentially create
issues if the SAM file is large and there is limited disk space. The
default alignment parameters are the same as PathoScope 2.0’s default
alignment parameters, but users can provide their own Bowtie 2 alignment
settings if desired.</p>
<p>We will now align the sample reads (reads.fastq) to the target
reference genomes using the Bowtie 2 indexes that we just built.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Create a temp directory to store output bam file</span></span>
<span><span class="va">output_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">output_temp</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get path to example reads</span></span>
<span><span class="va">readPath</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"reads.fastq"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Align reads to the target genomes</span></span>
<span><span class="va">target_map</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/align_target_bowtie.html">align_target_bowtie</a></span><span class="op">(</span></span>
<span>    read1 <span class="op">=</span> <span class="va">readPath</span>,</span>
<span>    lib_dir <span class="op">=</span> <span class="va">index_temp</span>,</span>
<span>    libs <span class="op">=</span> <span class="st">"target"</span>,</span>
<span>    align_dir <span class="op">=</span> <span class="va">output_temp</span>,</span>
<span>    align_file <span class="op">=</span> <span class="st">"bowtie_target"</span>,</span>
<span>    overwrite <span class="op">=</span> <span class="cn">TRUE</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## [1] "Samtools found on system. Using samtools to create bam file"</span></span></code></pre>
<pre><code><span><span class="co">## arguments 'show.output.on.console', 'minimized' and 'invisible' are for Windows only</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="metafilter">MetaFilter<a class="anchor" aria-label="anchor" href="#metafilter"></a>
</h3>
<p>The last step in the mapping workflow is to filter the output BAM
file according to the reference genome for the filter/host species.
Although we have already filtered out any unmapped reads, which may
belong to one or more host species or otherwise, there may still remain
some sort of unwelcome contamination in the data from the filter species
which we wish to remove. To do this, we employ
<code><a href="../reference/filter_host_bowtie.html">filter_host_bowtie()</a></code> (Rsubread equivalent:
<code><a href="../reference/filter_host.html">filter_host()</a></code>), which takes as an input the location of the
BAM file created from <code><a href="../reference/align_target_bowtie.html">align_target_bowtie()</a></code>, the directory
where the indexes are stored, and the names of the filter indexes to
align against, to produce a sorted BAM file with any reads that match
the filter libraries removed. We will then use this final BAM file
downstream for further analysis.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">final_map</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/filter_host_bowtie.html">filter_host_bowtie</a></span><span class="op">(</span></span>
<span>    reads_bam <span class="op">=</span> <span class="va">target_map</span>,</span>
<span>    lib_dir <span class="op">=</span> <span class="va">index_temp</span>,</span>
<span>    libs <span class="op">=</span> <span class="st">"filter"</span>,</span>
<span>    make_bam <span class="op">=</span> <span class="cn">TRUE</span>, <span class="co"># Set to true to create BAM output</span></span>
<span>    <span class="co"># Default is to create simplified .csv.gz output</span></span>
<span>    <span class="co"># The .csv.gz output is much quicker to create!</span></span>
<span>    overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    threads <span class="op">=</span> <span class="fl">1</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span></span>
<span><span class="co">## [1] "Samtools found on system. Using samtools to create bam file"</span></span></code></pre>
<pre><code><span><span class="co">## arguments 'show.output.on.console', 'minimized' and 'invisible' are for Windows only</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="evaluating-alignments-supplemental">Evaluating alignments (supplemental)<a class="anchor" aria-label="anchor" href="#evaluating-alignments-supplemental"></a>
</h3>
<p><strong>Note: the next two code blocks are included for the sake of
examining the vignette example, but are not useful for “real life” data.
To continue your analysis, head down to the Genome Identification
header.</strong></p>
<p>Prior to the last step in the analysis, we will look at the primary
alignments of the mapped reads in the filtered BAM file that we just
created using the <code><a href="../reference/filter_host_bowtie.html">filter_host_bowtie()</a></code> function. According
to the <a href="http://bowtie-bio.sourceforge.net/bowtie2/manual.shtml" class="external-link">Bowtie2
manual</a>, a primary alignment is described as the alignment that
received the highest alignment score among all alignments for that read.
When looking at the primary alignments of the mapped reads, we can see
that the majority of reads have mapped to the correct <em>Staphylococcus
aureus</em> RF122 strain. However, some residual reads have primary
alignments to the other <em>S. aureus</em> strains which are incorrect.
If we were to stop the analysis at this point, we could potentially be
lead to believe that our sample has increased microbial diversity, when
it actually does not.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bamFile</span> <span class="op">&lt;-</span> <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html" class="external-link">BamFile</a></span><span class="op">(</span><span class="va">final_map</span><span class="op">)</span></span>
<span></span>
<span><span class="va">param</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html" class="external-link">ScanBamParam</a></span><span class="op">(</span></span>
<span>    flag <span class="op">=</span> <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html" class="external-link">scanBamFlag</a></span><span class="op">(</span>isSecondaryAlignment <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>    what <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"flag"</span>, <span class="st">"rname"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="co">#Gets info about primary alignments</span></span>
<span></span>
<span><span class="va">aln</span> <span class="op">&lt;-</span> <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/scanBam.html" class="external-link">scanBam</a></span><span class="op">(</span><span class="va">bamFile</span>, param <span class="op">=</span> <span class="va">param</span><span class="op">)</span></span>
<span><span class="va">accession_all</span> <span class="op">&lt;-</span> <span class="va">aln</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">rname</span></span>
<span><span class="va">unique_accession_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">accession_all</span><span class="op">)</span></span>
<span><span class="va">accession_all_inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">accession_all</span>, <span class="va">unique_accession_all</span><span class="op">)</span></span>
<span><span class="va">unique_accession_genome_name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span></span>
<span>  <span class="fu">taxize</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/taxize/reference/genbank2uid.html" class="external-link">genbank2uid</a></span><span class="op">(</span><span class="va">unique_accession_all</span>,</span>
<span>                      batch_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unique_accession_all</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"name"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="va">unique_accession_genome_name</span><span class="op">[</span><span class="va">accession_all_inds</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">',.*'</span>, <span class="st">''</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(ST398).*"</span>, <span class="st">"\\1"</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(N315).*"</span>, <span class="st">"\\1"</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(Newman).*"</span>, <span class="st">"\\1"</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(Mu3).*"</span>, <span class="st">"\\1"</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(Mu50).*"</span>, <span class="st">"\\1"</span>, <span class="va">.</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(RF122).*"</span>, <span class="st">"\\1"</span>, <span class="va">.</span><span class="op">)</span></span>
<span><span class="va">read_count_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">genome_name_all</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>  <span class="va">read_count_table</span>,</span>
<span>  col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Genome Assigned"</span>, <span class="st">"Read Count"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Genome Assigned</th>
<th align="right">Read Count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Staphylococcus aureus RF122</td>
<td align="right">805</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus aureus subsp. aureus ST398</td>
<td align="right">51</td>
</tr>
<tr class="odd">
<td align="left">Staphylococcus aureus subsp. aureus str. Newman</td>
<td align="right">48</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus aureus subsp. aureus Mu3</td>
<td align="right">47</td>
</tr>
<tr class="odd">
<td align="left">Staphylococcus aureus subsp. aureus N315</td>
<td align="right">22</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus aureus subsp. aureus Mu50</td>
<td align="right">18</td>
</tr>
</tbody>
</table>
<p>We can also look at the secondary alignments of the mapped reads
within our filtered BAM file. A secondary alignment occurs when a read
maps to multiple different genomes. We can see that the majority of our
secondary alignments are to the other Staphylococcus aureus strains,
which makes sense considering that the majority of the primary
alignments were to the correct Staphylococcus aureus RF122 strain.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">bamFile</span> <span class="op">&lt;-</span> <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/BamFile-class.html" class="external-link">BamFile</a></span><span class="op">(</span><span class="va">final_map</span><span class="op">)</span></span>
<span></span>
<span><span class="va">param</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html" class="external-link">ScanBamParam</a></span><span class="op">(</span></span>
<span>    flag <span class="op">=</span> <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/ScanBamParam-class.html" class="external-link">scanBamFlag</a></span><span class="op">(</span>isSecondaryAlignment <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>,</span>
<span>    what <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"flag"</span>, <span class="st">"rname"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="co">#Gets info about secondary alignments</span></span>
<span></span>
<span><span class="va">aln</span> <span class="op">&lt;-</span> <span class="fu">Rsamtools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Rsamtools/man/scanBam.html" class="external-link">scanBam</a></span><span class="op">(</span><span class="va">bamFile</span>, param <span class="op">=</span> <span class="va">param</span><span class="op">)</span></span>
<span><span class="va">accession_all</span> <span class="op">&lt;-</span> <span class="va">aln</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span><span class="op">$</span><span class="va">rname</span></span>
<span><span class="va">unique_accession_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">accession_all</span><span class="op">)</span></span>
<span><span class="va">accession_all_inds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="va">accession_all</span>, <span class="va">unique_accession_all</span><span class="op">)</span></span>
<span><span class="va">unique_accession_taxid</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressMessages</a></span><span class="op">(</span></span>
<span>    <span class="fu">taxize</span><span class="fu">::</span><span class="fu"><a href="https://docs.ropensci.org/taxize/reference/genbank2uid.html" class="external-link">genbank2uid</a></span><span class="op">(</span><span class="va">unique_accession_all</span>,</span>
<span>                        batch_size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">unique_accession_all</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">unique_accession_genome_name</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="va">unique_accession_taxid</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/attr.html" class="external-link">attr</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"name"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="va">unique_accession_genome_name</span><span class="op">[</span><span class="va">accession_all_inds</span><span class="op">]</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">',.*'</span>, <span class="st">''</span>, <span class="va">genome_name_all</span><span class="op">)</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(ST398).*"</span>, <span class="st">"\\1"</span>, <span class="va">genome_name_all</span><span class="op">)</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(N315).*"</span>, <span class="st">"\\1"</span>, <span class="va">genome_name_all</span><span class="op">)</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(Newman).*"</span>, <span class="st">"\\1"</span>, <span class="va">genome_name_all</span><span class="op">)</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(Mu3).*"</span>, <span class="st">"\\1"</span>, <span class="va">genome_name_all</span><span class="op">)</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(Mu50).*"</span>, <span class="st">"\\1"</span>, <span class="va">genome_name_all</span><span class="op">)</span></span>
<span><span class="va">genome_name_all</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">"(RF122).*"</span>, <span class="st">"\\1"</span>, <span class="va">genome_name_all</span><span class="op">)</span></span>
<span><span class="va">read_count_table</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="va">genome_name_all</span><span class="op">)</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span></span>
<span>  <span class="va">read_count_table</span>,</span>
<span>  col.names <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Genome Assigned"</span>, <span class="st">"Read Count"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">Genome Assigned</th>
<th align="right">Read Count</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Staphylococcus aureus subsp. aureus ST398</td>
<td align="right">1264</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus aureus subsp. aureus str. Newman</td>
<td align="right">852</td>
</tr>
<tr class="odd">
<td align="left">Staphylococcus aureus subsp. aureus Mu3</td>
<td align="right">732</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus aureus subsp. aureus Mu50</td>
<td align="right">270</td>
</tr>
<tr class="odd">
<td align="left">Staphylococcus aureus subsp. aureus N315</td>
<td align="right">235</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus aureus RF122</td>
<td align="right">90</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="metaid-origin-genome-identification">MetaID: Origin Genome Identification<a class="anchor" aria-label="anchor" href="#metaid-origin-genome-identification"></a>
</h2>
<p>Following the proper alignment of a sample to all target and filter
libraries of interest, we may proceed in identifying which genomes are
most likely to be represented in the sample. This identification
workflow is the core of MetaScope; it features a Bayesian read
reassignment model which dramatically improves specificity and
sensitivity over other methods (Francis et. al 2013). This is because
such a method identifies reads with unique alignments and uses them to
guide the reassignment of reads with ambiguous alignments.</p>
<p>The MetaID identification module consists of a single function,
<code>MetaScope_ID()</code>, which reads in a .bam file, annotates the
taxonomy and genome names, reduces the mapping ambiguity using a mixture
model, and outputs a .csv file with the results. Currently, it assumes
that the genome library/.bam files use NCBI accession names for
reference names.</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/metascope_id.html">metascope_id</a></span><span class="op">(</span></span>
<span>  <span class="va">final_map</span>,</span>
<span>  input_type <span class="op">=</span> <span class="st">"bam"</span>,</span>
<span>  <span class="co"># change input_type to "csv.gz" when not creating a BAM</span></span>
<span>  aligner <span class="op">=</span> <span class="st">"bowtie2"</span>,</span>
<span>  num_species_plot <span class="op">=</span> <span class="fl">0</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="va">output</span>,</span>
<span>             format <span class="op">=</span> <span class="st">"html"</span>,</span>
<span>             digits <span class="op">=</span> <span class="fl">2</span>,</span>
<span>             caption <span class="op">=</span> <span class="st">"Table of MetaScope ID results"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
Table of MetaScope ID results
</caption>
<thead><tr>
<th style="text-align:left;">
TaxonomyID
</th>
<th style="text-align:left;">
Genome
</th>
<th style="text-align:right;">
read_count
</th>
<th style="text-align:right;">
Proportion
</th>
<th style="text-align:right;">
readsEM
</th>
<th style="text-align:right;">
EMProportion
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
273036
</td>
<td style="text-align:left;">
Staphylococcus aureus RF122, complete sequence
</td>
<td style="text-align:right;">
865
</td>
<td style="text-align:right;">
0.87
</td>
<td style="text-align:right;">
862.3
</td>
<td style="text-align:right;">
0.87
</td>
</tr>
<tr>
<td style="text-align:left;">
418127
</td>
<td style="text-align:left;">
Staphylococcus aureus subsp. aureus Mu3, complete sequence
</td>
<td style="text-align:right;">
114
</td>
<td style="text-align:right;">
0.12
</td>
<td style="text-align:right;">
115.0
</td>
<td style="text-align:right;">
0.12
</td>
</tr>
<tr>
<td style="text-align:left;">
426430
</td>
<td style="text-align:left;">
Staphylococcus aureus subsp. aureus str. Newman, complete sequence
</td>
<td style="text-align:right;">
12
</td>
<td style="text-align:right;">
0.01
</td>
<td style="text-align:right;">
13.6
</td>
<td style="text-align:right;">
0.01
</td>
</tr>
</tbody>
</table>
<p><strong>Note: the next code block is included for the sake of
examining the vignette example, but can be skipped. Your results are in
the CSV file produced by the <code><a href="../reference/metascope_id.html">metascope_id()</a></code>
function.</strong></p>
<p>We will now look at the read reassignment results reported in the
output CSV file.</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">relevant_col</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">dirname</a></span><span class="op">(</span><span class="va">final_map</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="st">"bowtie_target.metascope_id.csv"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.csv</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fl">2</span><span class="op">:</span><span class="fl">4</span><span class="op">)</span></span>
<span></span>
<span><span class="va">relevant_col</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    Genome <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">Genome</span>, <span class="st">',.*'</span>, <span class="st">''</span><span class="op">)</span>,</span>
<span>    Genome <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">Genome</span>, <span class="st">"(ST398).*"</span>, <span class="st">"\\1"</span><span class="op">)</span>,</span>
<span>    Genome <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">Genome</span>, <span class="st">"(N315).*"</span>, <span class="st">"\\1"</span><span class="op">)</span>,</span>
<span>    Genome <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">Genome</span>, <span class="st">"(Newman).*"</span>, <span class="st">"\\1"</span><span class="op">)</span>,</span>
<span>    Genome <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">Genome</span>, <span class="st">"(Mu3).*"</span>, <span class="st">"\\1"</span><span class="op">)</span>,</span>
<span>    Genome <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">Genome</span>, <span class="st">"(RF122).*"</span>, <span class="st">"\\1"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="68%">
<col width="15%">
<col width="15%">
</colgroup>
<thead><tr class="header">
<th align="left">Genome</th>
<th align="right">read_count</th>
<th align="right">Proportion</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Staphylococcus aureus RF122</td>
<td align="right">865</td>
<td align="right">0.8728557</td>
</tr>
<tr class="even">
<td align="left">Staphylococcus aureus subsp. aureus Mu3</td>
<td align="right">114</td>
<td align="right">0.1150353</td>
</tr>
<tr class="odd">
<td align="left">Staphylococcus aureus subsp. aureus str. Newman</td>
<td align="right">12</td>
<td align="right">0.0121090</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="st">".bowtie2.cerr.txt"</span><span class="op">)</span></span></code></pre></div>
<p>We can see that the read reassignment function has reassigned the
majority of the ambiguous alignments back to the <em>Staphylococcus
aureus</em> RF122 strain, the correct strain of origin.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: AlmaLinux 8.10 (Cerulean Leopard)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: FlexiBLAS NETLIB;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] magrittr_2.0.3   MetaScope_1.7.0  BiocStyle_2.32.1</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] tidyselect_1.2.1        dplyr_1.1.4             blob_1.2.4             </span></span>
<span><span class="co">##  [4] bitops_1.0-9            urltools_1.7.3          filelock_1.0.3         </span></span>
<span><span class="co">##  [7] Biostrings_2.72.1       fastmap_1.2.0           BiocFileCache_2.12.0   </span></span>
<span><span class="co">## [10] digest_0.6.37           lifecycle_1.0.4         RSQLite_2.3.7          </span></span>
<span><span class="co">## [13] compiler_4.4.0          rlang_1.1.4             sass_0.4.9             </span></span>
<span><span class="co">## [16] tools_4.4.0             utf8_1.2.4              yaml_2.3.10            </span></span>
<span><span class="co">## [19] data.table_1.16.0       knitr_1.48              conditionz_0.1.0       </span></span>
<span><span class="co">## [22] htmlwidgets_1.6.4       bit_4.5.0               curl_5.2.3             </span></span>
<span><span class="co">## [25] xml2_1.3.6              BiocParallel_1.38.0     httpcode_0.3.0         </span></span>
<span><span class="co">## [28] withr_3.0.1             purrr_1.0.2             BiocGenerics_0.50.0    </span></span>
<span><span class="co">## [31] desc_1.4.3              triebeard_0.4.1         grid_4.4.0             </span></span>
<span><span class="co">## [34] stats4_4.4.0            fansi_1.0.6             iterators_1.0.14       </span></span>
<span><span class="co">## [37] crul_1.5.0              taxize_0.9.100.1        cli_3.6.3              </span></span>
<span><span class="co">## [40] rmarkdown_2.28          crayon_1.5.3            ragg_1.3.0             </span></span>
<span><span class="co">## [43] generics_0.1.3          rstudioapi_0.16.0       httr_1.4.7             </span></span>
<span><span class="co">## [46] DBI_1.2.3               ape_5.8                 cachem_1.1.0           </span></span>
<span><span class="co">## [49] stringr_1.5.1           zlibbioc_1.50.0         parallel_4.4.0         </span></span>
<span><span class="co">## [52] BiocManager_1.30.22     XVector_0.44.0          vctrs_0.6.5            </span></span>
<span><span class="co">## [55] Matrix_1.7-0            jsonlite_1.8.9          bookdown_0.39          </span></span>
<span><span class="co">## [58] IRanges_2.38.1          S4Vectors_0.42.1        bit64_4.5.2            </span></span>
<span><span class="co">## [61] systemfonts_1.0.6       foreach_1.5.2           jquerylib_0.1.4        </span></span>
<span><span class="co">## [64] glue_1.8.0              Rbowtie2_2.10.0         pkgdown_2.0.9          </span></span>
<span><span class="co">## [67] codetools_0.2-20        stringi_1.8.4           GenomeInfoDb_1.40.1    </span></span>
<span><span class="co">## [70] GenomicRanges_1.56.1    UCSC.utils_1.0.0        tibble_3.2.1           </span></span>
<span><span class="co">## [73] pillar_1.9.0            htmltools_0.5.8.1       GenomeInfoDbData_1.2.12</span></span>
<span><span class="co">## [76] R6_2.5.1                dbplyr_2.5.0            textshaping_0.3.7      </span></span>
<span><span class="co">## [79] evaluate_1.0.0          lattice_0.22-6          highr_0.11             </span></span>
<span><span class="co">## [82] Rsamtools_2.20.0        memoise_2.0.1           bslib_0.8.0            </span></span>
<span><span class="co">## [85] Rcpp_1.0.13             uuid_1.2-1              nlme_3.1-164           </span></span>
<span><span class="co">## [88] xfun_0.45               fs_1.6.4                zoo_1.8-12             </span></span>
<span><span class="co">## [91] pkgconfig_2.0.3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
