<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="MetaScope">
<title>Condensed MetaScope Tutorial • MetaScope</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Condensed MetaScope Tutorial">
<meta property="og:description" content="MetaScope">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a>
    <a class="dropdown-item" href="../articles/MetaScopeTutorial.html">Condensed MetaScope Tutorial</a>
    <a class="dropdown-item" href="../articles/SILVA.html">Usage with SILVA 16S Database</a>
    <a class="dropdown-item" href="../articles/Example_Script.html">Example MetaScope Script</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/wejlab/MetaScope">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Condensed MetaScope Tutorial</h1>
                        <h4 data-toc-skip class="author">Aubrey
Odom</h4>
            <address class="author_afil">
      Program in Bioinformatics, Boston University, Boston,
MA<br><a class="author_email" href="mailto:#"></a><a href="mailto:aodom@bu.edu" class="email">aodom@bu.edu</a>
      </address>
                  
            <h4 data-toc-skip class="date">June 20, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/vignettes/MetaScopeTutorial.Rmd" class="external-link"><code>vignettes/MetaScopeTutorial.Rmd</code></a></small>
      <div class="d-none name"><code>MetaScopeTutorial.Rmd</code></div>
    </div>

    
    
<p>This is a condensed tutorial for the MetaScope package. We will walk
through the individual steps to perform an analysis. A thorough
explanation of the MetaScope package is available in the <a href="https://wejlab.github.io/metascope-docs/articles/MetaScope_vignette.html">MetaScope
Vignette</a>.</p>
<div class="section level2">
<h2 id="overview">Overview<a class="anchor" aria-label="anchor" href="#overview"></a>
</h2>
<p>The following diagram reveals the general module workflow of the
MetaScope package.</p>
<div class="float">
<img src="MetaScope%20Workflow.png" alt="Figure 1. The MetaScope workflow and its associated modules with function descriptions. The MetaRef, MetaAlign, MetaFilter, and MetaID modules form the backbone of package operation, whereas the MetaDemultiplex, MetaBLAST, and MetaCombine modules are complementary to the core package functionality."><div class="figcaption">Figure 1. The MetaScope workflow and its
associated modules with function descriptions. The MetaRef, MetaAlign,
MetaFilter, and MetaID modules form the backbone of package operation,
whereas the MetaDemultiplex, MetaBLAST, and MetaCombine modules are
complementary to the core package functionality.</div>
</div>
<p>The core modules are as follows: 1. <strong>MetaDemultiplex:</strong>
Obtain non-barcoded sequencing reads 2. <strong>MetaRef:</strong> Obtain
target and filter genome sequences from NCBI nucleotide database and
index using a given aligner 3. <strong>MetaAlign:</strong> Align
sequencing reads to indexed target genome sequences 4.
<strong>MetaFilter:</strong> Remove reads mapped to indexed host genome
sequences 5. <strong>MetaID:</strong> Reassign ambiguously mapped reads
to probable source genome 6. <strong>MetaBLAST:</strong> BLAST assigned
reads against the NCBI nucleotide database to check identity 7.
<strong>MetaCombine:</strong> Aggregate samples into a
MultiAssayExperiment compatible with the <code>animalcules</code> R
package.</p>
<p>There are two sub-workflows that are included in the package, as seen
in Figure 1: the Rbowtie2 and the Rsubread workflow. The major
difference is that the functions in the MetaRef, MetaAlign, and
MetaFilter modules differ by the aligner utilized.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>In order to install MetaScope from GitHub, run the following
code:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"devtools"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"wejlab/MetaScope"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="op">{</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/wejlab/metascope" class="external-link">MetaScope</a></span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="entrez-key">Entrez key<a class="anchor" aria-label="anchor" href="#entrez-key"></a>
</h3>
<p>We highly recommend grabbing an Entrez key using the function
<code><a href="https://docs.ropensci.org/taxize/reference/key_helpers.html" class="external-link">taxize::use_entrez()</a></code>. Then set your key as a global
environment variable like so:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">NCBI_key</span> <span class="op">&lt;-</span> <span class="st">"&lt;Your key here&gt;"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/options.html" class="external-link">options</a></span><span class="op">(</span><span class="st">"ENTREZ_KEY"</span> <span class="op">=</span> <span class="va">NCBI_key</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="metademultiplex-optional">MetaDemultiplex (optional)<a class="anchor" aria-label="anchor" href="#metademultiplex-optional"></a>
</h2>
<p>This is an optional step in the analysis process depending on whether
your reads are multiplexed. The reads which we are currently trying to
analyze are not multiplexed and therefore this step is skipped in our
analysis. The example shown below is using different reads that are
barcoded in order to show the utility of the function.</p>
<div class="section level3">
<h3 id="running-meta_demultiplex">Running <code>meta_demultiplex</code><a class="anchor" aria-label="anchor" href="#running-meta_demultiplex"></a>
</h3>
<p>You will need the following files: 1. <strong>barcodeFile</strong>:
Path to a file containing a .tsv matrix with a header row, and then
sample names (column 1) and barcodes (column 2).
(<code>barcodePath</code>) 2. <strong>indexFile</strong>: Path to a
.fastq file that contains the barcodes for each read. The headers should
be the same (and in the same order) as readFile, and the sequence in the
indexFile should be the corresponding barcode for each read. Quality
scores are not considered. (<code>indexPath</code>) 3.
<strong>readFile</strong>: Path to the sequencing read .fastq file that
corresponds to the indexFile.</p>
<p>Additional parameters can be examined by accessing the
<code>meta_demultiplex</code> documentation with
<code><a href="../reference/meta_demultiplex.html">?meta_demultiplex</a></code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Get barcode, index, and read data locations</span></span>
<span><span class="va">barcodePath</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"barcodes.txt"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">indexPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example_index.fastq"</span>,</span>
<span>                         package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span><span class="va">readPath</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example.fastq"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Get barcode, index, and read data locations</span></span>
<span><span class="va">demult</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/meta_demultiplex.html">meta_demultiplex</a></span><span class="op">(</span><span class="va">barcodePath</span>,</span>
<span>                   <span class="va">indexPath</span>,</span>
<span>                   <span class="va">readPath</span>,</span>
<span>                   rcBarcodes <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                   hammingDist <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                   location <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">demult</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="output">Output<a class="anchor" aria-label="anchor" href="#output"></a>
</h3>
<p>The <code>meta_demultiplex</code> function will returns multiple
.fastq files that contain all reads whose index matches the barcodes
given. These files will be written to the location directory, and will
be named based on the given sampleNames and barcodes,
e.g. ‘./demultiplex_fastq/SampleName1_GGAATTATCGGT.fastq.gz’</p>
</div>
</div>
<div class="section level2">
<h2 id="metaref-reference-genome-library">MetaRef: Reference Genome Library<a class="anchor" aria-label="anchor" href="#metaref-reference-genome-library"></a>
</h2>
<div class="section level3">
<h3 id="downloading-ncbi-nucleotide-genomes">Downloading NCBI nucleotide genomes<a class="anchor" aria-label="anchor" href="#downloading-ncbi-nucleotide-genomes"></a>
</h3>
<p>Most users will need to download a library of reference genomes if
they are not using their own library. This can be completed using the
<code>download_refseq</code> function. The following code chunks are not
run as they are quite large. This step is best completed by running a
background computing job as it may take some time.</p>
<div class="section level4">
<h4 id="reference-versus-representive-genomes">Reference versus Representive genomes<a class="anchor" aria-label="anchor" href="#reference-versus-representive-genomes"></a>
</h4>
<p>The <code>download_refseq</code> function allows users to download
reference genomes from NCBI’s Reference Sequence (RefSeq) database and
the larger nucleotide database if desired. Reference genomes are defined
by the NCBI as high-quality, well-annotated genome sequence that serves
as a representative example for a particular species or organism.</p>
<p>A complete rundown of the differentiation between these categories is
described on the NCBI website: <a href="https://www.ncbi.nlm.nih.gov/refseq/about/prokaryotes/" class="external-link uri">https://www.ncbi.nlm.nih.gov/refseq/about/prokaryotes/</a></p>
</div>
<div class="section level4">
<h4 id="s-analyses">16S Analyses<a class="anchor" aria-label="anchor" href="#s-analyses"></a>
</h4>
<p>For a 16S amplicon sequencing microbiome analysis, you will only need
bacterial genomes. You can download all bacterial genomes from the NCBI
RefSeq database like so:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># For RefSeq data, set reference = TRUE and representative = FALSE</span></span>
<span><span class="fu"><a href="../reference/download_refseq.html">download_refseq</a></span><span class="op">(</span><span class="st">'bacteria'</span>, reference <span class="op">=</span> <span class="cn">TRUE</span>, representative <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                out_dir <span class="op">=</span> <span class="cn">NULL</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, patho_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># If the RefSeq assortment is too limited, set representative = TRUE</span></span>
<span><span class="fu"><a href="../reference/download_refseq.html">download_refseq</a></span><span class="op">(</span><span class="st">'bacteria'</span>, reference <span class="op">=</span> <span class="cn">TRUE</span>, representative <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                out_dir <span class="op">=</span> <span class="cn">NULL</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, patho_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="metagenomics-whole-genome-sequencing">Metagenomics / Whole Genome Sequencing<a class="anchor" aria-label="anchor" href="#metagenomics-whole-genome-sequencing"></a>
</h4>
<div class="section level5">
<h5 id="target-genomes">Target genomes<a class="anchor" aria-label="anchor" href="#target-genomes"></a>
</h5>
<p>If you are performing a metagenomics analysis, you will likely want
to analyze fungi, bacteria, and viruses. The example below is assuming
that only reference genomes will be downloaded</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">taxa</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'bacteria'</span>, <span class="st">'viruses'</span>, <span class="st">'fungi'</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="../reference/download_refseq.html">download_refseq</a></span><span class="op">(</span><span class="va">taxa</span>, reference <span class="op">=</span> <span class="cn">TRUE</span>, representative <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                out_dir <span class="op">=</span> <span class="cn">NULL</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, patho_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
</div>
<div class="section level5">
<h5 id="host-genomes">Host genomes<a class="anchor" aria-label="anchor" href="#host-genomes"></a>
</h5>
<p>Given our ‘target’ genomes of interest in a metagenomics analysis we
likely want to filter out any reads belonging to the host on which the
sample was taken. So, in the instance that your host is human, you can
download the reference genome using:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/download_refseq.html">download_refseq</a></span><span class="op">(</span><span class="st">"Homo sapiens"</span>, reference <span class="op">=</span> <span class="cn">TRUE</span>, representative <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                out_dir <span class="op">=</span> <span class="cn">NULL</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, patho_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level4">
<h4 id="handpicked-reference-libraries">Handpicked Reference Libraries<a class="anchor" aria-label="anchor" href="#handpicked-reference-libraries"></a>
</h4>
<p>If you want to only use specific genomes in your libraries, you
should first check the included taxonomy table to check that your
organisms of interest are accessible. The taxonomy table object is
stored as <code>MetaScope:::taxonomy_table</code>. To search for your
organisms of interest, you can use any standard technique that you would
utilize to search for a table entry. For example, you can
use<code>View(MetaScope:::taxonomy_table)</code> and filter the table in
RStudio, or some other means of filtering the data.</p>
<p>The taxonomy table has the following structure:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">MetaScope</span><span class="fu">:::</span><span class="va">taxonomy_table</span><span class="op">)</span></span></code></pre></div>
<p>If I wanted to use only <em>Staphylococcus</em> bacterial genomes in
this table then I could filter the taxonomy table like so:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">all_staph_strains</span> <span class="op">&lt;-</span> <span class="fu">MetaScope</span><span class="fu">:::</span><span class="va">taxonomy_table</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">genus</span> <span class="op">==</span> <span class="st">"Staphylococcus"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">strain</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Download representative genomes</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">all_staph_strains</span>,</span>
<span>       <span class="kw">function</span><span class="op">(</span><span class="va">strain</span><span class="op">)</span> <span class="fu"><a href="../reference/download_refseq.html">download_refseq</a></span><span class="op">(</span><span class="va">strain</span>, reference <span class="op">=</span> <span class="cn">TRUE</span>, representative <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                out_dir <span class="op">=</span> <span class="cn">NULL</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, patho_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="providing-your-own-reference-library">Providing Your Own Reference Library<a class="anchor" aria-label="anchor" href="#providing-your-own-reference-library"></a>
</h3>
<p>Databases outside of the NCBI nucleotide database are acceptable
inputs as long as they are presented in .fasta format; the SILVA
database is one such example. Please see the MetaScope tutorial, [Using
the SILVA 16S rRNA Database with MetaScope][<a href="https://wejlab.github.io/metascope-docs/articles/SILVA.html" class="uri">https://wejlab.github.io/metascope-docs/articles/SILVA.html</a>].</p>
</div>
<div class="section level3">
<h3 id="creating-indices">Creating indices<a class="anchor" aria-label="anchor" href="#creating-indices"></a>
</h3>
<p>Given your host and filter genomes, the next step is to create
Bowtie- or Subread-compatible indices for the alignment modules.</p>
<div class="section level4">
<h4 id="bowtie-2-indices">Bowtie 2 indices<a class="anchor" aria-label="anchor" href="#bowtie-2-indices"></a>
</h4>
<p>Be sure to indicate the folder location of your indices
(<code>ref_dir</code>). The folder where your indices should be output
is given to the <code>lib_dir</code> parameter. Finally, the library
name of your indices (the individual files rather than folder locations)
is given to <code>lib_name</code>. The number of parallel computing
threads is given to <code>threads</code>.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nThreads</span> <span class="op">&lt;-</span> <span class="fl">1</span></span>
<span></span>
<span><span class="co"># Example - target genome fungi</span></span>
<span><span class="fu"><a href="../reference/mk_bowtie_index.html">mk_bowtie_index</a></span><span class="op">(</span>ref_dir <span class="op">=</span> <span class="st">"fungi"</span>, lib_dir <span class="op">=</span> <span class="st">"all_indices"</span>, </span>
<span>                lib_name <span class="op">=</span> <span class="st">"fungi"</span>, threads <span class="op">=</span> <span class="va">nThreads</span>, overwrite <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<p>In the case that there are unresolved errors occurring unrelated to
finding files or folders, we recommend troubleshooting by</p>
<ol style="list-style-type: decimal">
<li>Adding the parameter
<code>bowtie2_build_options = "--large-index"</code>
</li>
<li>Using the commandline Bowtie index creation tool,
<code>Bowtie2-build</code>
</li>
</ol>
</div>
<div class="section level4">
<h4 id="subread-indices">Subread indices<a class="anchor" aria-label="anchor" href="#subread-indices"></a>
</h4>
</div>
</div>
</div>
<div class="section level2">
<h2 id="metaalign">MetaAlign<a class="anchor" aria-label="anchor" href="#metaalign"></a>
</h2>
<p>We next perform the alignment step. Target genomes from reference
libraries will now be aligned to sample reads to create a BAM file.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># If unpaired, see documentation</span></span>
<span><span class="va">readPath1</span> <span class="op">&lt;-</span> <span class="st">"reads1.paired.fastq.gz"</span></span>
<span><span class="va">readPath2</span> <span class="op">&lt;-</span> <span class="st">"reads2.paired.fastq.gz"</span></span>
<span></span>
<span><span class="co"># Change to your target library names</span></span>
<span><span class="va">targets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"fungi"</span>, <span class="st">"bacteria"</span>, <span class="st">"viruses"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">alignment_directory</span> <span class="op">&lt;-</span> <span class="st">"&lt;where should alignments be output?&gt;"</span></span>
<span></span>
<span><span class="va">threads</span> <span class="op">&lt;-</span> <span class="fl">8</span></span>
<span></span>
<span><span class="co"># Usually expTag is a dynamically changing sample name</span></span>
<span><span class="va">expTag</span> <span class="op">&lt;-</span> <span class="st">"sampleX"</span></span>
<span></span>
<span><span class="co"># Location of alignment indices</span></span>
<span><span class="va">all_indices</span> <span class="op">&lt;-</span> <span class="st">"&lt;Location of alignment indices&gt;"</span></span></code></pre></div>
<div class="section level3">
<h3 id="bowtie-2">Bowtie 2<a class="anchor" aria-label="anchor" href="#bowtie-2"></a>
</h3>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"bt2_params"</span><span class="op">)</span></span>
<span><span class="co"># Use 16S params instead if you have 16S data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"bt2_16S_params"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">target_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/align_target_bowtie.html">align_target_bowtie</a></span><span class="op">(</span>read1 <span class="op">=</span> <span class="va">readPath1</span>,</span>
<span>                                  read2 <span class="op">=</span> <span class="va">readPath2</span>,</span>
<span>                                  <span class="co"># Where are indices stored?</span></span>
<span>                                  lib_dir <span class="op">=</span> <span class="va">all_indices</span>,</span>
<span>                                  libs <span class="op">=</span>  <span class="va">targets</span>,</span>
<span>                                  align_dir <span class="op">=</span> <span class="va">alignment_directory</span>,</span>
<span>                                  align_file <span class="op">=</span> <span class="va">expTag</span>,</span>
<span>                                  overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                  threads <span class="op">=</span> <span class="va">threads</span>,</span>
<span>                                  <span class="co"># Use 16S params if you have 16S data</span></span>
<span>                                  bowtie2_options <span class="op">=</span> <span class="va">bt2_params</span>,</span>
<span>                                  quiet <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="subread">Subread<a class="anchor" aria-label="anchor" href="#subread"></a>
</h3>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"align_details"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">target_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/align_target.html">align_target</a></span><span class="op">(</span>read1 <span class="op">=</span> <span class="va">readPath1</span>,</span>
<span>                           read2 <span class="op">=</span> <span class="va">readPath2</span>,</span>
<span>                           lib_dir <span class="op">=</span> <span class="va">all_indices</span>,</span>
<span>                           libs <span class="op">=</span> <span class="va">targets</span>,</span>
<span>                           threads <span class="op">=</span> <span class="va">threads</span>,</span>
<span>                           lib_dir <span class="op">=</span> <span class="va">target_ref_temp</span>,</span>
<span>                           align_file <span class="op">=</span> </span>
<span>                             <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">alignment_directory</span>, <span class="va">expTag</span><span class="op">)</span>,</span>
<span>                           subread_options <span class="op">=</span> <span class="va">align_details</span>,</span>
<span>                           quiet <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="metafilter">MetaFilter<a class="anchor" aria-label="anchor" href="#metafilter"></a>
</h2>
<p>We will now filter out host reads using the host reference genome
indices.</p>
<div class="section level3">
<h3 id="s-samples">16S Samples<a class="anchor" aria-label="anchor" href="#s-samples"></a>
</h3>
<p>When processing a 16S rRNA amplicon sequencing sample, the MetaFilter
step can be skipped entirely. You may proceed to the MetaID module.</p>
</div>
<div class="section level3">
<h3 id="metagenomics-samples">Metagenomics Samples<a class="anchor" aria-label="anchor" href="#metagenomics-samples"></a>
</h3>
<p>For either Bowtie 2 or Subread alignments, the primary input file
will be the alignment output by the MetaAlign module.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Change to your filter library name(s)</span></span>
<span><span class="va">filters</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"homo_sapiens"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">threads</span> <span class="op">&lt;-</span> <span class="fl">8</span></span>
<span></span>
<span><span class="va">output</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">alignment_directory</span>, <span class="va">expTag</span><span class="op">)</span>, <span class="st">"filtered"</span>, sep <span class="op">=</span> <span class="st">"."</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="bowtie-2-1">Bowtie 2<a class="anchor" aria-label="anchor" href="#bowtie-2-1"></a>
</h4>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"bt2_params"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># target_map is the bam file output by the MetaAlign module</span></span>
<span><span class="va">final_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_host_bowtie.html">filter_host_bowtie</a></span><span class="op">(</span>reads_bam <span class="op">=</span> <span class="va">target_map</span>,</span>
<span>                                lib_dir <span class="op">=</span> <span class="va">all_indices</span>,</span>
<span>                                libs <span class="op">=</span> <span class="va">filters</span>,</span>
<span>                                make_bam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                output <span class="op">=</span> <span class="va">output</span>,</span>
<span>                                threads <span class="op">=</span> <span class="va">threads</span>,</span>
<span>                                bowtie2_options <span class="op">=</span> <span class="va">bt2_params</span>,</span>
<span>                                overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                quiet <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                                bowtie2_options <span class="op">=</span> <span class="va">bt2_params</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="subread-1">Subread<a class="anchor" aria-label="anchor" href="#subread-1"></a>
</h4>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"align_details"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">final_map</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/filter_host.html">filter_host</a></span><span class="op">(</span></span>
<span>  reads_bam <span class="op">=</span> <span class="va">target_map</span>,</span>
<span>  lib_dir <span class="op">=</span> <span class="va">all_indices</span>,</span>
<span>  make_bam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  libs <span class="op">=</span> <span class="va">filters</span>,</span>
<span>  output <span class="op">=</span> <span class="va">output</span>,</span>
<span>  threads <span class="op">=</span> <span class="va">threads</span>,</span>
<span>  subread_options <span class="op">=</span> <span class="va">align_details</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="metaid">MetaID<a class="anchor" aria-label="anchor" href="#metaid"></a>
</h2>
<p>The final step is to reassign ambiguously mapped reads to their
likely source genome. This step is highly dependent on your reference
library used, so please check the documentation or the <a href="https://wejlab.github.io/metascope-docs/articles/SILVA.html">SILVA
tutorial</a> if you used a library other than RefSeq or NCBI
nucleotide.</p>
<p>Note that we highly recommend creating your own ENTREZ key, as noted
earlier in this tutorial.</p>
<div class="section level3">
<h3 id="bowtie-2-2">Bowtie 2<a class="anchor" aria-label="anchor" href="#bowtie-2-2"></a>
</h3>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">your_ENTREZ_KEY</span> <span class="op">&lt;-</span> <span class="st">"&lt;Your key here&gt;"</span></span>
<span><span class="va">outDir</span> <span class="op">&lt;-</span> <span class="st">"&lt;Your output directory&gt;"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>ENTREZ_KEY <span class="op">=</span> <span class="va">your_ENTREZ_KEY</span><span class="op">)</span></span></code></pre></div>
<div class="section level4">
<h4 id="s-sample">16S Sample<a class="anchor" aria-label="anchor" href="#s-sample"></a>
</h4>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># NOTE: If 16S sample, use MetaAlign output</span></span>
<span><span class="va">input_file</span> <span class="op">&lt;-</span> <span class="va">target_map</span></span>
<span></span>
<span><span class="fu"><a href="../reference/metascope_id.html">metascope_id</a></span><span class="op">(</span><span class="va">input_file</span>, </span>
<span>             input_type <span class="op">=</span> <span class="st">"bam"</span>,</span>
<span>             aligner <span class="op">=</span> <span class="st">"bowtie2"</span>,</span>
<span>             NCBI_key <span class="op">=</span> <span class="va">your_ENTREZ_KEY</span>,</span>
<span>             num_species_plot <span class="op">=</span> <span class="fl">15</span>,</span>
<span>             quiet <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             out_dir <span class="op">=</span> <span class="va">outDir</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="metagenomics-sample">Metagenomics Sample<a class="anchor" aria-label="anchor" href="#metagenomics-sample"></a>
</h4>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># NOTE: If Metagenomics sample, use MetaFilter output</span></span>
<span><span class="va">input_file</span> <span class="op">&lt;-</span> <span class="va">final_map</span></span>
<span></span>
<span><span class="fu"><a href="../reference/metascope_id.html">metascope_id</a></span><span class="op">(</span><span class="va">input_file</span>, </span>
<span>             input_type <span class="op">=</span> <span class="st">"csv.gz"</span>,</span>
<span>             aligner <span class="op">=</span> <span class="st">"bowtie2"</span>,</span>
<span>             NCBI_key <span class="op">=</span> <span class="va">your_ENTREZ_KEY</span>,</span>
<span>             num_species_plot <span class="op">=</span> <span class="fl">15</span>,</span>
<span>             quiet <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             out_dir <span class="op">=</span> <span class="va">outDir</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="subread-2">Subread<a class="anchor" aria-label="anchor" href="#subread-2"></a>
</h3>
<div class="section level4">
<h4 id="s-sample-1">16S Sample<a class="anchor" aria-label="anchor" href="#s-sample-1"></a>
</h4>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># NOTE: If 16S sample, use MetaAlign output</span></span>
<span><span class="va">input_file</span> <span class="op">&lt;-</span> <span class="va">target_map</span></span>
<span></span>
<span><span class="fu"><a href="../reference/metascope_id.html">metascope_id</a></span><span class="op">(</span><span class="va">input_file</span>, </span>
<span>             input_type <span class="op">=</span> <span class="st">"bam"</span>,</span>
<span>             aligner <span class="op">=</span> <span class="st">"subread"</span>,</span>
<span>             NCBI_key <span class="op">=</span> <span class="va">your_ENTREZ_KEY</span>,</span>
<span>             num_species_plot <span class="op">=</span> <span class="fl">15</span>,</span>
<span>             quiet <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             out_dir <span class="op">=</span> <span class="va">outDir</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="metagenomics-sample-1">Metagenomics sample<a class="anchor" aria-label="anchor" href="#metagenomics-sample-1"></a>
</h4>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># NOTE: If Metagenomics sample, use MetaFilter output</span></span>
<span><span class="va">input_file</span> <span class="op">&lt;-</span> <span class="va">final_map</span></span>
<span></span>
<span><span class="fu"><a href="../reference/metascope_id.html">metascope_id</a></span><span class="op">(</span><span class="va">input_file</span>, </span>
<span>             input_type <span class="op">=</span> <span class="st">"csv.gz"</span>,</span>
<span>             aligner <span class="op">=</span> <span class="st">"subread"</span>,</span>
<span>             NCBI_key <span class="op">=</span> <span class="va">your_ENTREZ_KEY</span>,</span>
<span>             num_species_plot <span class="op">=</span> <span class="fl">15</span>,</span>
<span>             quiet <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>             out_dir <span class="op">=</span> <span class="va">outDir</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
</div>
<div class="section level2">
<h2 id="metacombine">MetaCombine<a class="anchor" aria-label="anchor" href="#metacombine"></a>
</h2>
<p>Assuming that you’ve processed several samples using MetaScope, you
can aggregate all of your samples using the MetaCombine module. In this
example, we assume our reference library was created with the MetaRef
module (NCBI-formatted). This module produces a Multi-Assay
Experiment</p>
<div class="section level3">
<h3 id="create-fake-samples">Create fake samples<a class="anchor" aria-label="anchor" href="#create-fake-samples"></a>
</h3>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tempfolder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">tempfolder</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Create three different samples</span></span>
<span><span class="va">samp_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"X123"</span>, <span class="st">"X456"</span>, <span class="st">"X789"</span><span class="op">)</span></span>
<span><span class="va">all_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tempfolder</span>,</span>
<span>                       <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">samp_names</span>, <span class="st">".csv"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">create_IDcsv</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">out_file</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">final_taxids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"273036"</span>, <span class="st">"418127"</span>, <span class="st">"11234"</span><span class="op">)</span></span>
<span>  <span class="va">final_genomes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>    <span class="st">"Staphylococcus aureus RF122, complete sequence"</span>,</span>
<span>    <span class="st">"Staphylococcus aureus subsp. aureus Mu3, complete sequence"</span>,</span>
<span>    <span class="st">"Measles virus, complete genome"</span><span class="op">)</span></span>
<span>  <span class="va">best_hit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">1050</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span></span>
<span>  <span class="va">proportion</span> <span class="op">&lt;-</span> <span class="va">best_hit</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">best_hit</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span></span>
<span>  <span class="va">EMreads</span> <span class="op">&lt;-</span> <span class="va">best_hit</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span></span>
<span>  <span class="va">EMprop</span> <span class="op">&lt;-</span> <span class="va">proportion</span> <span class="op">+</span> <span class="fl">0.003</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>TaxonomyID <span class="op">=</span> <span class="va">final_taxids</span>,</span>
<span>                Genome <span class="op">=</span> <span class="va">final_genomes</span>,</span>
<span>                read_count <span class="op">=</span> <span class="va">best_hit</span>, Proportion <span class="op">=</span> <span class="va">proportion</span>,</span>
<span>                EMreads <span class="op">=</span> <span class="va">EMreads</span>, EMProportion <span class="op">=</span> <span class="va">EMprop</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">read_count</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>    <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">out_file</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Done!"</span><span class="op">)</span></span>
<span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">out_file</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">out_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="va">all_files</span>, <span class="va">create_IDcsv</span>, FUN.VALUE <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="create-annotation-data-for-samples">Create annotation data for samples<a class="anchor" aria-label="anchor" href="#create-annotation-data-for-samples"></a>
</h3>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">annot_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tempfolder</span>, <span class="st">"annot.csv"</span><span class="op">)</span></span>
<span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="va">samp_names</span>, RSV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pos"</span>, <span class="st">"neg"</span>, <span class="st">"pos"</span><span class="op">)</span>,</span>
<span>              month <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"March"</span>, <span class="st">"July"</span>, <span class="st">"Aug"</span><span class="op">)</span>,</span>
<span>              yrsold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.6</span>, <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">annot_dat</span>,</span>
<span>                   row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="convert-samples-to-mae">Convert samples to MAE<a class="anchor" aria-label="anchor" href="#convert-samples-to-mae"></a>
</h3>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">outMAE</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/convert_animalcules.html">convert_animalcules</a></span><span class="op">(</span>meta_counts <span class="op">=</span> <span class="va">out_files</span>,</span>
<span>                              annot_path <span class="op">=</span> <span class="va">annot_dat</span>,</span>
<span>                              which_annot_col <span class="op">=</span> <span class="st">"Sample"</span>,</span>
<span>                              end_string <span class="op">=</span> <span class="st">".metascope_id.csv"</span>,</span>
<span>                              qiime_biom_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                              NCBI_key <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">tempfolder</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="metablast">MetaBLAST<a class="anchor" aria-label="anchor" href="#metablast"></a>
</h2>
<p>An example using the MetaBLAST module is forthcoming.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.4.0 (2024-04-24)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu</span></span>
<span><span class="co">## Running under: AlmaLinux 8.10 (Cerulean Leopard)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: FlexiBLAS NETLIB;  LAPACK version 3.11.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: America/New_York</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats     graphics  grDevices utils     datasets  methods   base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] BiocStyle_2.32.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] vctrs_0.6.5         cli_3.6.2           knitr_1.47         </span></span>
<span><span class="co">##  [4] rlang_1.1.4         xfun_0.45           purrr_1.0.2        </span></span>
<span><span class="co">##  [7] textshaping_0.3.7   jsonlite_1.8.8      htmltools_0.5.8.1  </span></span>
<span><span class="co">## [10] ragg_1.3.0          sass_0.4.9          rmarkdown_2.27     </span></span>
<span><span class="co">## [13] evaluate_0.24.0     jquerylib_0.1.4     fastmap_1.2.0      </span></span>
<span><span class="co">## [16] yaml_2.3.8          lifecycle_1.0.4     memoise_2.0.1      </span></span>
<span><span class="co">## [19] bookdown_0.39       BiocManager_1.30.22 compiler_4.4.0     </span></span>
<span><span class="co">## [22] fs_1.6.4            htmlwidgets_1.6.4   rstudioapi_0.16.0  </span></span>
<span><span class="co">## [25] systemfonts_1.0.6   digest_0.6.35       R6_2.5.1           </span></span>
<span><span class="co">## [28] magrittr_2.0.3      bslib_0.7.0         tools_4.4.0        </span></span>
<span><span class="co">## [31] pkgdown_2.0.9       cachem_1.1.0        desc_1.4.3</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
