<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This is the main MetaScope target library mapping function, using Rsubread
and multiple libraries. Aligns to each library separately, filters unmapped
reads from each file, and then merges and sorts the .bam files from each
library into one output file. If desired, output can be passed to
`filter_host()` to remove reads that also map to filter library genomes."><title>Align microbiome reads to a set of reference libraries — align_target • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Align microbiome reads to a set of reference libraries — align_target"><meta property="og:description" content="This is the main MetaScope target library mapping function, using Rsubread
and multiple libraries. Aligns to each library separately, filters unmapped
reads from each file, and then merges and sorts the .bam files from each
library into one output file. If desired, output can be passed to
`filter_host()` to remove reads that also map to filter library genomes."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/MetaScope_vignette.html">Tutorial</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/compbiomed/MetaScope">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Align microbiome reads to a set of reference libraries</h1>
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/MetaScope/blob/HEAD/R/align_target.R" class="external-link"><code>R/align_target.R</code></a></small>
      <div class="d-none name"><code>align_target.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This is the main MetaScope target library mapping function, using Rsubread
and multiple libraries. Aligns to each library separately, filters unmapped
reads from each file, and then merges and sorts the .bam files from each
library into one output file. If desired, output can be passed to
`filter_host()` to remove reads that also map to filter library genomes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">align_target</span><span class="op">(</span></span>
<span>  <span class="va">read1</span>,</span>
<span>  read2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  lib_dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">libs</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  align_file <span class="op">=</span> <span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/fileutils.html" class="external-link">file_path_sans_ext</a></span><span class="op">(</span><span class="va">read1</span><span class="op">)</span>,</span>
<span>  subread_options <span class="op">=</span> <span class="va">align_details</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>read1</dt>
<dd><p>Path to the .fastq file to align.</p></dd>


<dt>read2</dt>
<dd><p>Optional: Location of the mate pair .fastq file to align.</p></dd>


<dt>lib_dir</dt>
<dd><p>Path to the index files for all libraries.</p></dd>


<dt>libs</dt>
<dd><p>A vector of character strings giving the basenames of the Subread
index files for alignment. If ALL indices to be used are located in the
current working directory, set <code>lib_dir = NULL</code>. Default is
<code>lib_dir = NULL</code>.</p></dd>


<dt>threads</dt>
<dd><p>The number of threads that can be utilized by the function.
Default is 1 thread.</p></dd>


<dt>align_file</dt>
<dd><p>The basename of the output alignment file (without trailing
.bam extension).</p></dd>


<dt>subread_options</dt>
<dd><p>A named <code>list</code> specifying alignment parameters
for the <code><a href="https://rdrr.io/pkg/Rsubread/man/align.html" class="external-link">Rsubread::align()</a></code> function, which is called inside
<code>align_target()</code>. Elements should include type, nthreads,
maxMismatches, nsubreads, phredOffset, unique, and nBestLocations.
Descriptions of these parameters are available under
<code><a href="https://rdrr.io/pkg/Rsubread/man/align.html" class="external-link">?Rsubread::align</a></code>. Defaults to the global <code>align_details</code>
object.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>This function writes a merged and sorted .bam file after aligning to
all reference libraries given, along with a summary report file, to the
user's working directory. The function also outputs the new .bam filename.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Align example reads to an example reference library using Rsubread</span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co">## Create temporary directory</span></span></span>
<span class="r-in"><span><span class="va">target_ref_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">target_ref_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Download genome</span></span></span>
<span class="r-in"><span><span class="va">tax</span> <span class="op">&lt;-</span> <span class="st">"Mycoplasma genitalium G37"</span></span></span>
<span class="r-in"><span><span class="va">all_ref</span> <span class="op">&lt;-</span> <span class="fu">MetaScope</span><span class="fu">::</span><span class="fu"><a href="download_refseq.html">download_refseq</a></span><span class="op">(</span><span class="va">tax</span>,</span></span>
<span class="r-in"><span>                                      reference <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                      representative <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                                      compress <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                                      out_dir <span class="op">=</span> <span class="va">target_ref_temp</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Finding Mycoplasma genitalium G37</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No ENTREZ API key provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Get one via taxize::use_entrez()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No ENTREZ API key provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Get one via taxize::use_entrez()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Mycoplasma genitalium G37 is a strain under the Bacteria Superkingdom</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Loading the refseq table for bacteria</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Creating table of relevant taxa</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Downloading 2 Mycoplasma genitalium G37 genome(s) from NCBI</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE! Downloaded 2 genomes to /scratch/141165.1.ood/RtmpwygBTw/file7c3324afa3a5//scratch/141165.1.ood/RtmpwygBTw/file7c3324afa3a5/Mycoplasma_genitalium_G37.fasta.gz</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create subread index</span></span></span>
<span class="r-in"><span><span class="va">ind_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="mk_subread_index.html">mk_subread_index</a></span><span class="op">(</span><span class="va">all_ref</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Library size is 0 GBs. Building the Rsubread index</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= setting ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                Index name : Mycoplasma_genitalium_G37                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Index space : base space                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Index split : no-split                                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Repeat threshold : 100 repeats                                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Gapped index : no                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||       Free / total memory : 200.5GB / 251.7GB                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Input files : 1 file in total                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                             o Mycoplasma_genitalium_G37.fasta.gz           ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Check the integrity of provided reference sequences ...                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || No format issues were found                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Scan uninformative subreads in reference sequences ...                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Estimate the index size...                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    8%,   0 mins elapsed, rate=491.0k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   16%,   0 mins elapsed, rate=914.5k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   24%,   0 mins elapsed, rate=1284.4k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   33%,   0 mins elapsed, rate=1594.4k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   41%,   0 mins elapsed, rate=1881.4k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   58%,   0 mins elapsed, rate=2450.5k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   66%,   0 mins elapsed, rate=2703.6k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   74%,   0 mins elapsed, rate=2944.2k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   83%,   0 mins elapsed, rate=3156.9k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   91%,   0 mins elapsed, rate=3337.6k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || 3.0 GB of memory is needed for index building.                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Build the index...                                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    8%,   0 mins elapsed, rate=49.5k bps/s                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   16%,   0 mins elapsed, rate=97.5k bps/s                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   24%,   0 mins elapsed, rate=144.4k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   33%,   0 mins elapsed, rate=190.0k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   41%,   0 mins elapsed, rate=234.3k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   58%,   0 mins elapsed, rate=321.4k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   66%,   0 mins elapsed, rate=363.0k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   74%,   0 mins elapsed, rate=403.8k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   83%,   0 mins elapsed, rate=443.5k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   91%,   0 mins elapsed, rate=482.3k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Save current index block...                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 0.0% finished ]                                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 10.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 20.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 30.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 40.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 50.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 60.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 70.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 80.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 90.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 100.0% finished ]                                                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Total running time: 0.2 minutes.                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||Index /scratch/141165.1.ood/RtmpwygBTw/file7c3324afa3a5/Mycoplasma_gen ... ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get path to example reads</span></span></span>
<span class="r-in"><span><span class="va">readPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"reads.fastq"</span>,</span></span>
<span class="r-in"><span>                        package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co">## Copy the example reads to the temp directory</span></span></span>
<span class="r-in"><span><span class="va">refPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">target_ref_temp</span>, <span class="st">"reads.fastq"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span>from <span class="op">=</span> <span class="va">readPath</span>, to <span class="op">=</span> <span class="va">refPath</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Modify alignment parameters object</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"align_details"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">align_details</span><span class="op">[[</span><span class="st">"type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"rna"</span></span></span>
<span class="r-in"><span><span class="va">align_details</span><span class="op">[[</span><span class="st">"phredOffset"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">50</span></span></span>
<span class="r-in"><span><span class="co"># Just to get it to align - toy example!</span></span></span>
<span class="r-in"><span><span class="va">align_details</span><span class="op">[[</span><span class="st">"maxMismatches"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">100</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Run alignment</span></span></span>
<span class="r-in"><span><span class="va">target_map</span> <span class="op">&lt;-</span> <span class="fu">align_target</span><span class="op">(</span><span class="va">refPath</span>,</span></span>
<span class="r-in"><span>                           libs <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">tax</span>, <span class="st">" "</span>, <span class="st">"_"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>                           lib_dir <span class="op">=</span> <span class="va">target_ref_temp</span>,</span></span>
<span class="r-in"><span>                           subread_options <span class="op">=</span> <span class="va">align_details</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Attempting to perform subread alignment on Mycoplasma_genitalium_G37 index</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.10.5</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= setting ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Function      : Read alignment (RNA-Seq)                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Input file    : reads.fastq                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Output file   : reads.Mycoplasma_genitalium_G37.bam (BAM)                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Index name    : Mycoplasma_genitalium_G37                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                    ------------------------------------                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                               Threads : 1                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                          Phred offset : 64                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                             Min votes : 3 / 10                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                        Max mismatches : 100                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Max indel length : 5                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||            Report multi-mapping reads : yes                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Max alignments per multi-mapping read : 16                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================ Running (28-Oct-2022 16:42:15, pid=31795) =================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Check the input reads.                                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || The input file contains base space reads.                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Initialise the memory objects.                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Estimate the mean read length.                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || The range of Phred scores observed in the data is [-14,-14]                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Create the output BAM file.                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Check the index.                                                           ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Init the voting space.                                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Global environment is initialised.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load the 1-th index block...                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || The index block has been loaded.                                           ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Start read mapping in chunk.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           Completed successfully.                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\====================================    ====================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================   Summary =================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Total reads : 1,500                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Mapped : 3 (0.2%)                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Uniquely mapped : 0                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Multi-mapping : 3                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                    Unmapped : 1,497                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Indels : 0                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                Running time : 1.0 minutes                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Filtering unmapped reads</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Library alignment complete</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> DONE! Alignments written to /scratch/141165.1.ood/RtmpwygBTw/file7c3324afa3a5/reads.bam</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Remove temporary folder</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">target_ref_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Site created by Aubrey Odom-Mabey.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

    </footer></div>

  

  

  </body></html>

