<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Align microbiome reads to set of indexed Bowtie2 libraries — align_target_bowtie • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Align microbiome reads to set of indexed Bowtie2 libraries — align_target_bowtie"><meta name="description" content="This is the main MetaScope target library mapping function, using Rbowtie2
and multiple libraries. Aligns to each library separately, filters unmapped
reads from each file, and then merges and sorts the .bam files from each
library into one output file. If desired, output can be passed to
`filter_host_bowtie()` to remove reads that also map to filter library
genomes."><meta property="og:description" content="This is the main MetaScope target library mapping function, using Rbowtie2
and multiple libraries. Aligns to each library separately, filters unmapped
reads from each file, and then merges and sorts the .bam files from each
library into one output file. If desired, output can be passed to
`filter_host_bowtie()` to remove reads that also map to filter library
genomes."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/publicly_hosted_indices_and_accessions.html">Publicly Hosted Indices</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Package News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/MetaScope"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Align microbiome reads to set of indexed Bowtie2 libraries</h1>
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/R/align_target.R" class="external-link"><code>R/align_target.R</code></a></small>
      <div class="d-none name"><code>align_target_bowtie.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This is the main MetaScope target library mapping function, using Rbowtie2
and multiple libraries. Aligns to each library separately, filters unmapped
reads from each file, and then merges and sorts the .bam files from each
library into one output file. If desired, output can be passed to
`filter_host_bowtie()` to remove reads that also map to filter library
genomes.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">align_target_bowtie</span><span class="op">(</span></span>
<span>  <span class="va">read1</span>,</span>
<span>  read2 <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">lib_dir</span>,</span>
<span>  <span class="va">libs</span>,</span>
<span>  <span class="va">align_dir</span>,</span>
<span>  <span class="va">align_file</span>,</span>
<span>  bowtie2_options <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  overwrite <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-read-">read1<a class="anchor" aria-label="anchor" href="#arg-read-"></a></dt>
<dd><p>Path to the .fastq file to align.</p></dd>


<dt id="arg-read-">read2<a class="anchor" aria-label="anchor" href="#arg-read-"></a></dt>
<dd><p>Optional: Location of the mate pair .fastq file to align.</p></dd>


<dt id="arg-lib-dir">lib_dir<a class="anchor" aria-label="anchor" href="#arg-lib-dir"></a></dt>
<dd><p>Path to the directory that contains the Bowtie2 indexes.</p></dd>


<dt id="arg-libs">libs<a class="anchor" aria-label="anchor" href="#arg-libs"></a></dt>
<dd><p>The basename of the Bowtie2 indexes to align against (without
trailing .bt2 or .bt2l extensions).</p></dd>


<dt id="arg-align-dir">align_dir<a class="anchor" aria-label="anchor" href="#arg-align-dir"></a></dt>
<dd><p>Path to the directory where the output alignment file should
be created.</p></dd>


<dt id="arg-align-file">align_file<a class="anchor" aria-label="anchor" href="#arg-align-file"></a></dt>
<dd><p>The basename of the output alignment file (without trailing
.bam extension).</p></dd>


<dt id="arg-bowtie-options">bowtie2_options<a class="anchor" aria-label="anchor" href="#arg-bowtie-options"></a></dt>
<dd><p>Optional: Additional parameters that can be passed to
the align_target_bowtie() function. To see all the available parameters use
Rbowtie2::bowtie2_usage(). See Details for default parameters. NOTE: Users
should pass all their parameters as one string and if optional parameters
are given then the user is responsible for entering all the parameters to
be used by Bowtie2. The only parameter that should NOT be specified here is
the number of threads.</p></dd>


<dt id="arg-threads">threads<a class="anchor" aria-label="anchor" href="#arg-threads"></a></dt>
<dd><p>The number of threads that can be utilized by the function.
Default is 1 thread.</p></dd>


<dt id="arg-overwrite">overwrite<a class="anchor" aria-label="anchor" href="#arg-overwrite"></a></dt>
<dd><p>Whether existing files should be overwritten. Default is
FALSE.</p></dd>


<dt id="arg-quiet">quiet<a class="anchor" aria-label="anchor" href="#arg-quiet"></a></dt>
<dd><p>Turns off most messages. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns the path to where the output alignment file is stored.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>The default parameters are the same that PathoScope 2.0 uses.
"–very-sensitive-local -k 100 –score-min L,20,1.0"</p>
<p>If you experience any issues with reading the input files, make sure that
the file(s) are not located in a read-only folder. This can be circumvented
by copying files to a new location before running the function.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Align example reads to an example reference library using Rbowtie2</span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co">## Create temporary directory to store file</span></span></span>
<span class="r-in"><span><span class="va">target_ref_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">target_ref_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">tmp_accession</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_accessions.sql"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Dowload reference genome</span></span></span>
<span class="r-in"><span><span class="fu">MetaScope</span><span class="fu">::</span><span class="fu"><a href="download_refseq.html">download_refseq</a></span><span class="op">(</span><span class="st">"Morbillivirus hominis"</span>,</span></span>
<span class="r-in"><span>                           reference <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                           representative <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                           compress <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                           out_dir <span class="op">=</span> <span class="va">target_ref_temp</span>,</span></span>
<span class="r-in"><span>                           caching <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                           accession_path <span class="op">=</span> <span class="va">tmp_accession</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/scratch/333180.1.ood/RtmpXcbbC1/file2ae601e30e239/Morbillivirus_hominis.fasta.gz"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create temporary directory to store the indices</span></span></span>
<span class="r-in"><span><span class="va">index_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">index_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create bowtie2 index</span></span></span>
<span class="r-in"><span><span class="fu">MetaScope</span><span class="fu">::</span><span class="fu"><a href="mk_bowtie_index.html">mk_bowtie_index</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  ref_dir <span class="op">=</span> <span class="va">target_ref_temp</span>,</span></span>
<span class="r-in"><span>  lib_dir <span class="op">=</span> <span class="va">index_temp</span>,</span></span>
<span class="r-in"><span>  lib_name <span class="op">=</span> <span class="st">"target"</span>,</span></span>
<span class="r-in"><span>  overwrite <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> arguments 'show.output.on.console', 'minimized' and 'invisible' are for Windows only</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Index building complete</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/scratch/333180.1.ood/RtmpXcbbC1/file2ae60542d3386"</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create temporary directory for final file</span></span></span>
<span class="r-in"><span><span class="va">output_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">output_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get path to example reads</span></span></span>
<span class="r-in"><span><span class="va">readPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example.fastq"</span>,</span></span>
<span class="r-in"><span>                        package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Align to target genomes</span></span></span>
<span class="r-in"><span><span class="va">target_map</span> <span class="op">&lt;-</span></span></span>
<span class="r-in"><span>  <span class="fu">MetaScope</span><span class="fu">::</span><span class="fu">align_target_bowtie</span><span class="op">(</span></span></span>
<span class="r-in"><span>    read1 <span class="op">=</span> <span class="va">readPath</span>,</span></span>
<span class="r-in"><span>    lib_dir <span class="op">=</span> <span class="va">index_temp</span>,</span></span>
<span class="r-in"><span>    libs <span class="op">=</span> <span class="st">"target"</span>,</span></span>
<span class="r-in"><span>    align_dir <span class="op">=</span> <span class="va">output_temp</span>,</span></span>
<span class="r-in"><span>    align_file <span class="op">=</span> <span class="st">"bowtie_target"</span>,</span></span>
<span class="r-in"><span>    overwrite <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    bowtie2_options <span class="op">=</span> <span class="st">"--very-sensitive-local"</span></span></span>
<span class="r-in"><span>  <span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "Samtools found on system. Using samtools to create bam file"</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> arguments 'show.output.on.console', 'minimized' and 'invisible' are for Windows only</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Remove extra folders</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">target_ref_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">index_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">output_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

