<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Create a multi-assay experiment from MetaScope output for usage with animalcules — convert_animalcules • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create a multi-assay experiment from MetaScope output for usage with animalcules — convert_animalcules"><meta name="description" content="Upon completion of the MetaScope pipeline, users can analyze and visualize
abundances in their samples using the animalcules package. This function
allows interoperability of metascope_id output with both animalcules
and QIIME. After running this function, the user should save the returned MAE
to an RDS file using a function like saveRDS to upload the output into
the animalcules package."><meta property="og:description" content="Upon completion of the MetaScope pipeline, users can analyze and visualize
abundances in their samples using the animalcules package. This function
allows interoperability of metascope_id output with both animalcules
and QIIME. After running this function, the user should save the returned MAE
to an RDS file using a function like saveRDS to upload the output into
the animalcules package."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/publicly_hosted_indices_and_accessions.html">Publicly Hosted Indices</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Package News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/MetaScope"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Create a multi-assay experiment from MetaScope output for usage with animalcules</h1>
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/R/convert_animalcules.R" class="external-link"><code>R/convert_animalcules.R</code></a></small>
      <div class="d-none name"><code>convert_animalcules.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Upon completion of the MetaScope pipeline, users can analyze and visualize
abundances in their samples using the animalcules package. This function
allows interoperability of <code>metascope_id</code> output with both animalcules
and QIIME. After running this function, the user should save the returned MAE
to an RDS file using a function like <code>saveRDS</code> to upload the output into
the <code>animalcules</code> package.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">convert_animalcules</span><span class="op">(</span></span>
<span>  <span class="va">meta_counts</span>,</span>
<span>  <span class="va">annot_path</span>,</span>
<span>  <span class="va">which_annot_col</span>,</span>
<span>  end_string <span class="op">=</span> <span class="st">".metascope_id.csv"</span>,</span>
<span>  qiime_biom_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  path_to_write <span class="op">=</span> <span class="st">"."</span>,</span>
<span>  accession_path <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-meta-counts">meta_counts<a class="anchor" aria-label="anchor" href="#arg-meta-counts"></a></dt>
<dd><p>A vector of filepaths to the counts ID CSVs output by
<code><a href="metascope_id.html">metascope_id()</a></code>.</p></dd>


<dt id="arg-annot-path">annot_path<a class="anchor" aria-label="anchor" href="#arg-annot-path"></a></dt>
<dd><p>The filepath to the CSV annotation file for the samples.
This CSV metadata/annotation file should contain at least two columns,
one with names of all samples WITHOUT the extension listed in <code>end_string</code>,
e.g. for output file "sample_x76.metascope_id.csv", the column specified in
<code>which_annot_col</code> should contain the entry "sample_x76". Sample names
containing characters "_", "-", and "." are fine, however sample names
beginning with numbers should be renamed to have a prefix, e.g. "777897sample"
should be renamed to "X777897sample" for both the output file name and the
annotation name.</p></dd>


<dt id="arg-which-annot-col">which_annot_col<a class="anchor" aria-label="anchor" href="#arg-which-annot-col"></a></dt>
<dd><p>The name of the column of the annotation file
containing the sample IDs. These should be the same as the
<code>meta_counts</code> root filenames.</p></dd>


<dt id="arg-end-string">end_string<a class="anchor" aria-label="anchor" href="#arg-end-string"></a></dt>
<dd><p>The end string used at the end of the metascope_id files.
Default is ".metascope_id.csv".</p></dd>


<dt id="arg-qiime-biom-out">qiime_biom_out<a class="anchor" aria-label="anchor" href="#arg-qiime-biom-out"></a></dt>
<dd><p>Would you also like a qiime-compatible biom file
output? If yes, two files will be saved: one is a biom file of the counts
table, and the other is a specifically formatted mapping file of metadata
information. Default is <code>FALSE</code>.</p></dd>


<dt id="arg-path-to-write">path_to_write<a class="anchor" aria-label="anchor" href="#arg-path-to-write"></a></dt>
<dd><p>If <code>qiime_biom_out = TRUE</code>, where should output QIIME
files be written? Should be a character string of the folder path. Default is
'.', i.e. the current working directory.</p></dd>


<dt id="arg-accession-path">accession_path<a class="anchor" aria-label="anchor" href="#arg-accession-path"></a></dt>
<dd><p>(character) Path to taxonomizr accessions. See
<code><a href="https://rdrr.io/pkg/taxonomizr/man/prepareDatabase.html" class="external-link">taxonomizr::prepareDatabase()</a></code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns a MultiAssay Experiment file of combined sample counts data
  and/or biom file and mapping file for analysis with QIIME. The MultiAssay
  Experiment will have a counts assay ("MGX").</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="va">tempfolder</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">tempfolder</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create three different samples</span></span></span>
<span class="r-in"><span><span class="va">samp_names</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"X123"</span>, <span class="st">"X456"</span>, <span class="st">"X789"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">all_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tempfolder</span>,</span></span>
<span class="r-in"><span>                       <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">samp_names</span>, <span class="st">".csv"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">create_IDcsv</span> <span class="op">&lt;-</span> <span class="kw">function</span> <span class="op">(</span><span class="va">out_file</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>  <span class="va">final_taxids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"273036"</span>, <span class="st">"418127"</span>, <span class="st">"11234"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">final_genomes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    <span class="st">"Staphylococcus aureus RF122, complete sequence"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"Staphylococcus aureus subsp. aureus Mu3, complete sequence"</span>,</span></span>
<span class="r-in"><span>    <span class="st">"Measles virus, complete genome"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">best_hit</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">100</span>, <span class="fl">1050</span><span class="op">)</span>, <span class="fl">3</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">proportion</span> <span class="op">&lt;-</span> <span class="va">best_hit</span><span class="op">/</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">best_hit</span><span class="op">)</span> <span class="op">|&gt;</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">EMreads</span> <span class="op">&lt;-</span> <span class="va">best_hit</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span>, <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="va">EMprop</span> <span class="op">&lt;-</span> <span class="va">proportion</span> <span class="op">+</span> <span class="fl">0.003</span></span></span>
<span class="r-in"><span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>TaxonomyID <span class="op">=</span> <span class="va">final_taxids</span>,</span></span>
<span class="r-in"><span>                Genome <span class="op">=</span> <span class="va">final_genomes</span>,</span></span>
<span class="r-in"><span>                read_count <span class="op">=</span> <span class="va">best_hit</span>, Proportion <span class="op">=</span> <span class="va">proportion</span>,</span></span>
<span class="r-in"><span>                EMreads <span class="op">=</span> <span class="va">EMreads</span>, EMProportion <span class="op">=</span> <span class="va">EMprop</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">.data</span><span class="op">$</span><span class="va">read_count</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>    <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">out_file</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">message</a></span><span class="op">(</span><span class="st">"Done!"</span><span class="op">)</span></span></span>
<span class="r-in"><span>  <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="va">out_file</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span><span class="va">out_files</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="va">all_files</span>, <span class="va">create_IDcsv</span>, FUN.VALUE <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Done!</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create annotation data for samples</span></span></span>
<span class="r-in"><span><span class="va">annot_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">tempfolder</span>, <span class="st">"annot.csv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/tibble.html" class="external-link">tibble</a></span><span class="op">(</span>Sample <span class="op">=</span> <span class="va">samp_names</span>, RSV <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pos"</span>, <span class="st">"neg"</span>, <span class="st">"pos"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>              month <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"March"</span>, <span class="st">"July"</span>, <span class="st">"Aug"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>              yrsold <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.6</span>, <span class="fl">0.2</span><span class="op">)</span><span class="op">)</span> <span class="op">|&gt;</span></span></span>
<span class="r-in"><span>  <span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span>file <span class="op">=</span> <span class="va">annot_dat</span>,</span></span>
<span class="r-in"><span>                   row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create temporary taxonomizr accession</span></span></span>
<span class="r-in"><span><span class="va">tmp_accession</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_accessions.sql"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Convert samples to MAE</span></span></span>
<span class="r-in"><span><span class="va">outMAE</span> <span class="op">&lt;-</span> <span class="fu">convert_animalcules</span><span class="op">(</span>meta_counts <span class="op">=</span> <span class="va">out_files</span>,</span></span>
<span class="r-in"><span>                              annot_path <span class="op">=</span> <span class="va">annot_dat</span>,</span></span>
<span class="r-in"><span>                              which_annot_col <span class="op">=</span> <span class="st">"Sample"</span>,</span></span>
<span class="r-in"><span>                              end_string <span class="op">=</span> <span class="st">".metascope_id.csv"</span>,</span></span>
<span class="r-in"><span>                              qiime_biom_out <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                              accession_path <span class="op">=</span> <span class="va">tmp_accession</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Looking up taxon UIDs in NCBI database</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">tempfolder</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

