<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function will automatically download RefSeq genome libraries in a fasta
format from the specified taxon. The function will first download the
summary report at:
ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/**kingdom**/assembly_summary.txt,
and then use this file to download the genome(s) and combine them in a
single compressed or uncompressed .fasta file."><title>Download RefSeq genome libraries — download_refseq • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.11/clipboard.min.js" integrity="sha512-7O5pXpc0oCRrxk8RUfDYFgn0nO1t+jLuIOQdOMRp4APB7uZ4vSjspzp5y6YDtDs4VzUSTbWzBFZ/LKJhnyFOKw==" crossorigin="anonymous" referrerpolicy="no-referrer"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Download RefSeq genome libraries — download_refseq"><meta property="og:description" content="This function will automatically download RefSeq genome libraries in a fasta
format from the specified taxon. The function will first download the
summary report at:
ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/**kingdom**/assembly_summary.txt,
and then use this file to download the genome(s) and combine them in a
single compressed or uncompressed .fasta file."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark" data-bs-theme="dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-tutorials">Tutorials</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-tutorials">
    <a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a>
    <a class="dropdown-item" href="../articles/MetaScopeTutorial.html">Condensed MetaScope Tutorial</a>
    <a class="dropdown-item" href="../articles/SILVA.html">Usage with SILVA 16S Database</a>
    <a class="dropdown-item" href="../articles/Example_Script.html">Example MetaScope Script</a>
  </div>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/wejlab/MetaScope">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Download RefSeq genome libraries</h1>
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/R/download_refseq.R" class="external-link"><code>R/download_refseq.R</code></a></small>
      <div class="d-none name"><code>download_refseq.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function will automatically download RefSeq genome libraries in a fasta
format from the specified taxon. The function will first download the
summary report at:
<code>ftp://ftp.ncbi.nlm.nih.gov/genomes/refseq/**kingdom**/assembly_summary.txt</code>,
and then use this file to download the genome(s) and combine them in a
single compressed or uncompressed .fasta file.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">download_refseq</span><span class="op">(</span></span>
<span>  <span class="va">taxon</span>,</span>
<span>  reference <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  representative <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  compress <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  patho_out <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  caching <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>taxon</dt>
<dd><p>Name of single taxon to download. The taxon name should be a
recognized NCBI scientific or common name, with no grammatical or
capitalization inconsistencies. All available taxonomies are visible by
accessing the <code>MetaScope:::taxonomy_table</code> object included in the package.</p></dd>


<dt>reference</dt>
<dd><p>Download only RefSeq reference genomes? Defaults to
<code>TRUE</code>. Automatically set to <code>TRUE</code> if <code>representative =
TRUE</code>.</p></dd>


<dt>representative</dt>
<dd><p>Download RefSeq representative and reference genomes?
Defaults to <code>FALSE</code>. If <code>TRUE</code>, reference is automatically set at
  <code>TRUE</code>.</p></dd>


<dt>compress</dt>
<dd><p>Compress the output .fasta file? Defaults to <code>TRUE</code>.</p></dd>


<dt>patho_out</dt>
<dd><p>Create duplicate outpute files compatible with PathoScope?
Defaults to <code>FALSE</code>.</p></dd>


<dt>out_dir</dt>
<dd><p>Character string giving the name of the directory to which
libraries should be output. Defaults to creation of a new temporary
directory.</p></dd>


<dt>caching</dt>
<dd><p>Whether to use BiocFileCache when downloading genomes.
Default is <code>FALSE</code>.</p></dd>


<dt>quiet</dt>
<dd><p>Turns off most messages. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>Returns a .fasta or .fasta.gz file of the desired RefSeq genomes.
This file is named after the kingdom selected and saved to the current
directory (e.g. 'bacteria.fasta.gz'). This function also has the option
to return a .fasta file formatted for PathoScope as well
(e.g. bacteria.pathoscope.fasta.gz') if <code>path_out = TRUE</code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>When selecting the <code>taxon</code> to be downloaded, if you receive an error
saying <code>Your input is not a valid taxon</code>, please take a look at the
<code>taxonomy_table</code> object, which can be accessed with the command
<code>MetaScope:::taxonomy_table)</code>. Only taxa with exact spelling as they appear
at any level of the table will be acknowledged.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Download RefSeq genomes</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Download all RefSeq reference Shotokuvirae kingdom genomes</span></span></span>
<span class="r-in"><span><span class="fu">download_refseq</span><span class="op">(</span><span class="st">'Shotokuvirae'</span>, reference <span class="op">=</span> <span class="cn">TRUE</span>, representative <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                out_dir <span class="op">=</span> <span class="cn">NULL</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>, patho_out <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                caching <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No ENTREZ API key provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Get one via taxize::use_entrez()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No ENTREZ API key provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Get one via taxize::use_entrez()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "/scratch/260630.1.ood/RtmpB7j73T/file2fa80f61d1a6c0/Shotokuvirae.fasta.gz"</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.9.</p>
</div>

    </footer></div>

  

  

  </body></html>

