<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Helper function for demultiplexing — extract_reads • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Helper function for demultiplexing — extract_reads"><meta name="description" content="Helper function for demultiplexing sequencing reads, designed in a way to
allow for parallelization across barcodes (parallel extraction of reads by
barcode). This function takes a specific barcode (numeric index) from lists
of sample names/barcodes, a Biostrings::DNAStringSet of barcodes by
sequence header, and a Biostrings::QualityScaledXStringSet of reads
corresponding to the barcodes. Based on the barcode index given, it extracts
all reads for the indexed barcode and writes all the reads from that barcode
to a separate .fastq file."><meta property="og:description" content="Helper function for demultiplexing sequencing reads, designed in a way to
allow for parallelization across barcodes (parallel extraction of reads by
barcode). This function takes a specific barcode (numeric index) from lists
of sample names/barcodes, a Biostrings::DNAStringSet of barcodes by
sequence header, and a Biostrings::QualityScaledXStringSet of reads
corresponding to the barcodes. Based on the barcode index given, it extracts
all reads for the indexed barcode and writes all the reads from that barcode
to a separate .fastq file."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/publicly_hosted_indices_and_accessions.html">Publicly Hosted Indices</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Package News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/MetaScope"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Helper function for demultiplexing</h1>
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/R/demultiplex.R" class="external-link"><code>R/demultiplex.R</code></a></small>
      <div class="d-none name"><code>extract_reads.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Helper function for demultiplexing sequencing reads, designed in a way to
allow for parallelization across barcodes (parallel extraction of reads by
barcode). This function takes a specific barcode (numeric index) from lists
of sample names/barcodes, a <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a></code> of barcodes by
sequence header, and a <code><a href="https://rdrr.io/pkg/Biostrings/man/QualityScaledXStringSet-class.html" class="external-link">Biostrings::QualityScaledXStringSet</a></code> of reads
corresponding to the barcodes. Based on the barcode index given, it extracts
all reads for the indexed barcode and writes all the reads from that barcode
to a separate .fastq file.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">extract_reads</span><span class="op">(</span></span>
<span>  <span class="va">barcodeIndex</span>,</span>
<span>  <span class="va">barcodes</span>,</span>
<span>  <span class="va">sampleNames</span>,</span>
<span>  <span class="va">index</span>,</span>
<span>  <span class="va">reads</span>,</span>
<span>  location <span class="op">=</span> <span class="st">"./demultiplex_fastq"</span>,</span>
<span>  rcBarcodes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  hDist <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-barcodeindex">barcodeIndex<a class="anchor" aria-label="anchor" href="#arg-barcodeindex"></a></dt>
<dd><p>Which barcode (integer number or index) in the barcodes
or sample name to use for read extraction.</p></dd>


<dt id="arg-barcodes">barcodes<a class="anchor" aria-label="anchor" href="#arg-barcodes"></a></dt>
<dd><p>A list of all barcodes in the sequencing dataset. Correlates
and in same order as <code>sampleNames</code>.</p></dd>


<dt id="arg-samplenames">sampleNames<a class="anchor" aria-label="anchor" href="#arg-samplenames"></a></dt>
<dd><p>A list of sample names or identifiers associated with each
barcode in the barcodes list.</p></dd>


<dt id="arg-index">index<a class="anchor" aria-label="anchor" href="#arg-index"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-class.html" class="external-link">Biostrings::DNAStringSet</a></code> that contains the read headers
and barcode sequence for each header in the sequence slot.</p></dd>


<dt id="arg-reads">reads<a class="anchor" aria-label="anchor" href="#arg-reads"></a></dt>
<dd><p>A <code><a href="https://rdrr.io/pkg/Biostrings/man/QualityScaledXStringSet-class.html" class="external-link">Biostrings::QualityScaledXStringSet</a></code> that has the same
headers and order as the index file, but contains the read sequences and
their quality scores.</p></dd>


<dt id="arg-location">location<a class="anchor" aria-label="anchor" href="#arg-location"></a></dt>
<dd><p>A directory location to store the demultiplexed read files.
Defaults to generate a new subdirectory at './demultiplex_fastq'</p></dd>


<dt id="arg-rcbarcodes">rcBarcodes<a class="anchor" aria-label="anchor" href="#arg-rcbarcodes"></a></dt>
<dd><p>Should the barcode indices in the barcodes list be reverse
complemented to match the sequences in the index DNAStringSet? Defaults to
<code>TRUE</code>.</p></dd>


<dt id="arg-hdist">hDist<a class="anchor" aria-label="anchor" href="#arg-hdist"></a></dt>
<dd><p>Uses a Hamming Distance or number of base differences to allow
for inexact matches for the barcodes/indexes. Defaults to 0. Warning: if
the Hamming Distance is &gt;=1 and this leads to inexact index matches to more
than one barcode, that read will be written to more than one demultiplexed
read files.</p></dd>


<dt id="arg-quiet">quiet<a class="anchor" aria-label="anchor" href="#arg-quiet"></a></dt>
<dd><p>Turns off most messages. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Writes a single .fastq file that contains all reads whose index
  matches the barcode specified. This file will be written to the location
  directory, and will be named based on the specified sampleName and barcode,
  e.g. './demultiplex_fastq/SampleName1_GGAATTATCGGT.fastq.gz'</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create temporary directory</span></span></span>
<span class="r-in"><span><span class="va">ref_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">ref_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Load example barcode, index, and read data into R session</span></span></span>
<span class="r-in"><span><span class="va">barcodePath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"barcodes.txt"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">bcFile</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/read.table.html" class="external-link">read.table</a></span><span class="op">(</span><span class="va">barcodePath</span>, sep <span class="op">=</span> <span class="st">"\t"</span>, header <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">indexPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example_index.fastq"</span>,</span></span>
<span class="r-in"><span>package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">inds</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/XStringSet-io.html" class="external-link">readDNAStringSet</a></span><span class="op">(</span><span class="va">indexPath</span>, format <span class="op">=</span> <span class="st">"fastq"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">readPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example.fastq"</span>,</span></span>
<span class="r-in"><span>                        package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">reads</span> <span class="op">&lt;-</span> <span class="fu">Biostrings</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/Biostrings/man/QualityScaledXStringSet-class.html" class="external-link">readQualityScaledDNAStringSet</a></span><span class="op">(</span><span class="va">readPath</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract reads from the first barcode</span></span></span>
<span class="r-in"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu">extract_reads</span><span class="op">(</span><span class="fl">1</span>, <span class="va">bcFile</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="va">bcFile</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">inds</span>, <span class="va">reads</span>,</span></span>
<span class="r-in"><span>                        rcBarcodes <span class="op">=</span> <span class="cn">FALSE</span>, location <span class="op">=</span> <span class="va">ref_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Extract reads from multiple barcodes</span></span></span>
<span class="r-in"><span><span class="va">more_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="va">extract_reads</span>, <span class="va">bcFile</span><span class="op">[</span>, <span class="fl">2</span><span class="op">]</span>, <span class="va">bcFile</span><span class="op">[</span>, <span class="fl">1</span><span class="op">]</span>, <span class="va">inds</span>,</span></span>
<span class="r-in"><span>                       <span class="va">reads</span>, rcBarcodes <span class="op">=</span> <span class="cn">FALSE</span>, location <span class="op">=</span> <span class="va">ref_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Remove temporary directory</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">ref_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

