<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Use Rsubread to align reads against one or more filter libraries and subsequently remove mapped reads — filter_host • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Use Rsubread to align reads against one or more filter libraries and subsequently remove mapped reads — filter_host"><meta name="description" content="After aligning your sample to a target library with align_target(),
use filter_host() to remove unwelcome host contamination using filter
reference libraries. This function takes as input the name of the .bam file
produced via align_target(), and produces a sorted .bam file with any
reads that match the filter libraries removed. This resulting .bam file may
be used upstream for further analysis. This function uses Rsubread. For the
Rbowtie2 equivalent of this function, see filter_host_bowtie."><meta property="og:description" content="After aligning your sample to a target library with align_target(),
use filter_host() to remove unwelcome host contamination using filter
reference libraries. This function takes as input the name of the .bam file
produced via align_target(), and produces a sorted .bam file with any
reads that match the filter libraries removed. This resulting .bam file may
be used upstream for further analysis. This function uses Rsubread. For the
Rbowtie2 equivalent of this function, see filter_host_bowtie."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.9.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/publicly_hosted_indices_and_accessions.html">Publicly Hosted Indices</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Package News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/MetaScope"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Use Rsubread to align reads against one or more filter libraries and subsequently remove mapped reads</h1>
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/R/filter_host.R" class="external-link"><code>R/filter_host.R</code></a></small>
      <div class="d-none name"><code>filter_host.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>After aligning your sample to a target library with <code><a href="align_target.html">align_target()</a></code>,
use <code>filter_host()</code> to remove unwelcome host contamination using filter
reference libraries. This function takes as input the name of the .bam file
produced via <code><a href="align_target.html">align_target()</a></code>, and produces a sorted .bam file with any
reads that match the filter libraries removed. This resulting .bam file may
be used upstream for further analysis. This function uses Rsubread. For the
Rbowtie2 equivalent of this function, see <code>filter_host_bowtie</code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">filter_host</span><span class="op">(</span></span>
<span>  <span class="va">reads_bam</span>,</span>
<span>  lib_dir <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  <span class="va">libs</span>,</span>
<span>  make_bam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  output <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/fileutils.html" class="external-link">file_path_sans_ext</a></span><span class="op">(</span><span class="va">reads_bam</span><span class="op">)</span>, <span class="st">"filtered"</span>, sep <span class="op">=</span> <span class="st">"."</span><span class="op">)</span>,</span>
<span>  subread_options <span class="op">=</span> <span class="va">align_details</span>,</span>
<span>  YS <span class="op">=</span> <span class="fl">1e+05</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-reads-bam">reads_bam<a class="anchor" aria-label="anchor" href="#arg-reads-bam"></a></dt>
<dd><p>The name of a merged, sorted .bam file that has previously
been aligned to a reference library. Likely, the output from running an
instance of <code><a href="align_target.html">align_target()</a></code>.</p></dd>


<dt id="arg-lib-dir">lib_dir<a class="anchor" aria-label="anchor" href="#arg-lib-dir"></a></dt>
<dd><p>Path to the directory that contains the filter Subread index
files.</p></dd>


<dt id="arg-libs">libs<a class="anchor" aria-label="anchor" href="#arg-libs"></a></dt>
<dd><p>The basename of the filter libraries (without index
extension).</p></dd>


<dt id="arg-make-bam">make_bam<a class="anchor" aria-label="anchor" href="#arg-make-bam"></a></dt>
<dd><p>Logical, whether to also output a bam file with host reads
filtered out. A .csv.gz file will be created instead if <code>FALSE</code>.
Creating a bam file is costly on resources over creating a compressed
csv file with only relevant information, so default is <code>FALSE</code>.</p></dd>


<dt id="arg-output">output<a class="anchor" aria-label="anchor" href="#arg-output"></a></dt>
<dd><p>The desired name of the output .bam or .csv.gz file. Extension is
automatically defined by whether <code>make_bam = TRUE.</code> Default is the
basename of <code>unfiltered_bam</code> + <code>.filtered</code> + extension.</p></dd>


<dt id="arg-subread-options">subread_options<a class="anchor" aria-label="anchor" href="#arg-subread-options"></a></dt>
<dd><p>A named <code>list</code> specifying alignment parameters
for the <code><a href="https://rdrr.io/pkg/Rsubread/man/align.html" class="external-link">Rsubread::align()</a></code> function, which is called inside
<code><a href="align_target.html">align_target()</a></code>. Elements should include type, nthreads,
maxMismatches, nsubreads, phredOffset, unique, and nBestLocations.
Descriptions of these parameters are available under
<code><a href="https://rdrr.io/pkg/Rsubread/man/align.html" class="external-link">?Rsubread::align</a></code>. Defaults to the global <code>align_details</code>
object.</p></dd>


<dt id="arg-ys">YS<a class="anchor" aria-label="anchor" href="#arg-ys"></a></dt>
<dd><p>yieldSize, an integer. The number of alignments to be read in from
the bam file at once for chunked functions. Default is 100000.</p></dd>


<dt id="arg-threads">threads<a class="anchor" aria-label="anchor" href="#arg-threads"></a></dt>
<dd><p>The amount of threads available for the function. Default is 1
thread.</p></dd>


<dt id="arg-quiet">quiet<a class="anchor" aria-label="anchor" href="#arg-quiet"></a></dt>
<dd><p>Turns off most messages. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>The name of a filtered, sorted .bam file written to the user's
  current working directory. Or, if <code>make_bam = FALSE</code>, a .csv.gz file
  containing a data frame of only requisite information to run
  <code><a href="metascope_id.html">metascope_id()</a></code>.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>A compressed .csv can be created to produce a smaller output file that is
created more efficiently and is still compatible with <code><a href="metascope_id.html">metascope_id()</a></code>.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Filter reads from bam file that align to any of the filter libraries</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Assuming a bam file has been created previously with align_target()</span></span></span>
<span class="r-in"><span><span class="co"># \donttest{</span></span></span>
<span class="r-in"><span><span class="co">## Create temporary directory</span></span></span>
<span class="r-in"><span><span class="va">filter_ref_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">filter_ref_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create temporary taxonomizr accession</span></span></span>
<span class="r-in"><span><span class="va">tmp_accession</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_accessions.sql"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Download filter genome</span></span></span>
<span class="r-in"><span><span class="va">all_species</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Staphylococcus aureus subsp. aureus str. Newman"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">all_ref</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="va">all_species</span>, <span class="fu">MetaScope</span><span class="fu">::</span><span class="va"><a href="download_refseq.html">download_refseq</a></span>,</span></span>
<span class="r-in"><span>                  reference <span class="op">=</span> <span class="cn">FALSE</span>, representative <span class="op">=</span> <span class="cn">FALSE</span>, compress <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>                  out_dir <span class="op">=</span> <span class="va">filter_ref_temp</span>, caching <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                  accession_path <span class="op">=</span> <span class="va">tmp_accession</span>,</span></span>
<span class="r-in"><span>                  FUN.VALUE <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">ind_out</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">vapply</a></span><span class="op">(</span><span class="va">all_ref</span>, <span class="va">mk_subread_index</span>, FUN.VALUE <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">character</a></span><span class="op">(</span><span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.20.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= setting ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                Index name : Staphylococcus_aureus_subsp._aureus_str._N ... ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Index space : base space                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Index split : no-split                                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||          Repeat threshold : 100 repeats                                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||              Gapped index : no                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||       Free / total memory : 216.8GB / 251.3GB                              ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Input files : 1 file in total                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                             o Staphylococcus_aureus_subsp._aureus_str._Newman.fasta.gz ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= Running ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Check the integrity of provided reference sequences ...                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || No format issues were found                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Scan uninformative subreads in reference sequences ...                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Estimate the index size...                                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    8%,   0 mins elapsed, rate=4373.9k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   16%,   0 mins elapsed, rate=5605.4k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   24%,   0 mins elapsed, rate=6213.0k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   33%,   0 mins elapsed, rate=6507.9k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   41%,   0 mins elapsed, rate=6707.5k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   49%,   0 mins elapsed, rate=6864.3k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   58%,   0 mins elapsed, rate=6984.5k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   66%,   0 mins elapsed, rate=7082.5k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   74%,   0 mins elapsed, rate=7208.8k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   83%,   0 mins elapsed, rate=7288.4k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   91%,   0 mins elapsed, rate=7346.8k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || 3.0 GB of memory is needed for index building.                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Build the index...                                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||    8%,   0 mins elapsed, rate=638.9k bps/s                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   16%,   0 mins elapsed, rate=1081.8k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   24%,   0 mins elapsed, rate=1407.1k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   33%,   0 mins elapsed, rate=1657.8k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   41%,   0 mins elapsed, rate=1851.9k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   49%,   0 mins elapsed, rate=2014.4k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   58%,   0 mins elapsed, rate=2145.0k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   66%,   0 mins elapsed, rate=2259.9k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   74%,   0 mins elapsed, rate=2372.9k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   83%,   0 mins elapsed, rate=2457.8k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||   91%,   0 mins elapsed, rate=2533.2k bps/s                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Save current index block...                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 0.0% finished ]                                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 10.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 20.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 30.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 40.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 50.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 60.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 70.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 80.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 90.0% finished ]                                                        ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||  [ 100.0% finished ]                                                       ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Total running time: 0.3 minutes.                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||Index /scratch/339945.1.ood/RtmpEfBZjq/file1a657b2c304201/Staphylococc ... ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get path to example reads</span></span></span>
<span class="r-in"><span><span class="va">readPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"subread_target.bam"</span>,</span></span>
<span class="r-in"><span>                        package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Copy the example reads to the temp directory</span></span></span>
<span class="r-in"><span><span class="va">refPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">filter_ref_temp</span>, <span class="st">"subread_target.bam"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span>from <span class="op">=</span> <span class="va">readPath</span>, to <span class="op">=</span> <span class="va">refPath</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span><span class="fu">utils</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"align_details"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">align_details</span><span class="op">[[</span><span class="st">"type"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="st">"rna"</span></span></span>
<span class="r-in"><span><span class="va">align_details</span><span class="op">[[</span><span class="st">"phredOffset"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">10</span></span></span>
<span class="r-in"><span><span class="co"># Just to get it to align - toy example!</span></span></span>
<span class="r-in"><span><span class="va">align_details</span><span class="op">[[</span><span class="st">"maxMismatches"</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fl">10</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Align and filter reads</span></span></span>
<span class="r-in"><span><span class="va">filtered_map</span> <span class="op">&lt;-</span> <span class="fu">filter_host</span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="va">refPath</span>, lib_dir <span class="op">=</span> <span class="va">filter_ref_temp</span>,</span></span>
<span class="r-in"><span>  libs <span class="op">=</span> <span class="fu">stringr</span><span class="fu">::</span><span class="fu"><a href="https://stringr.tidyverse.org/reference/str_replace.html" class="external-link">str_replace_all</a></span><span class="op">(</span><span class="va">all_species</span>, <span class="st">" "</span>, <span class="st">"_"</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>  threads <span class="op">=</span> <span class="fl">1</span>, subread_options <span class="op">=</span> <span class="va">align_details</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========     _____ _    _ ____  _____  ______          _____  </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         =====         / ____| |  | |  _ \|  __ \|  ____|   /\   |  __ \ </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           =====      | (___ | |  | | |_) | |__) | |__     /  \  | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>             ====      \___ \| |  | |  _ &lt;|  _  /|  __|   / /\ \ | |  | |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>               ====    ____) | |__| | |_) | | \ \| |____ / ____ \| |__| |</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         ==========   |_____/ \____/|____/|_|  \_\______/_/    \_\_____/</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        Rsubread 2.20.0</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================= setting ==================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Function      : Read alignment (RNA-Seq)                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Input file    : intermediate.fastq.gz                                      ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Output file   : subread_target.Staphylococcus_aureus_subsp._aureus_str ... ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Index name    : Staphylococcus_aureus_subsp._aureus_str._Newman            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                    ------------------------------------                    ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                               Threads : 1                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                          Phred offset : 64                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                             Min votes : 3 / 10                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                        Max mismatches : 10                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Max indel length : 5                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||            Report multi-mapping reads : yes                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Max alignments per multi-mapping read : 16                                 ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //=============== Running (25-Apr-2025 18:51:23, pid=1729915) ================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Check the input reads.                                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || The input file contains base space reads.                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Initialise the memory objects.                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Estimate the mean read length.                                             ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Create the output BAM file.                                                ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Check the index.                                                           ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Init the voting space.                                                     ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Global environment is initialised.                                         ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Load the 1-th index block...                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || The index block has been loaded.                                           ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> || Start read mapping in chunk.                                               ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                           Completed successfully.                          ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\====================================    ====================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> //================================   Summary =================================\\</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                 Total reads : 1                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Mapped : 1 (100.0%)                                   ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||             Uniquely mapped : 0                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||               Multi-mapping : 1                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                    Unmapped : 0                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                      Indels : 0                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                Running time : 0.7 minutes                                  ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> ||                                                                            ||</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> \\============================================================================//</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   |                                                                              |                                                                      |   0%  |                                                                              |======================================================================| 100%</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Remove temporary directory</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">filter_ref_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="co"># }</span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

