<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.9.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/publicly_hosted_indices_and_accessions.html">Publicly Hosted Indices</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Package News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/MetaScope"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="metademultiplex-optional-">MetaDemultiplex (optional)<a class="anchor" aria-label="anchor" href="#metademultiplex-optional-"></a></h2>

      <div class="section-desc"><p>Obtain non-barcoded sequencing reads.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="meta_demultiplex.html">meta_demultiplex()</a></code>

        </dt>
        <dd>Demultiplexing sequencing reads</dd>
      </dl></div><div class="section level2">
      <h2 id="metaref">MetaRef<a class="anchor" aria-label="anchor" href="#metaref"></a></h2>

      <div class="section-desc"><p>Obtain target and filter genome sequences from NCBI nucleotide database and index using the Bowtie 2 or Subread aligner</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="download_refseq.html">download_refseq()</a></code>

        </dt>
        <dd>Download RefSeq genome libraries</dd>
      </dl><dl><dt>

          <code><a href="mk_bowtie_index.html">mk_bowtie_index()</a></code>

        </dt>
        <dd>Make a Bowtie2 index</dd>
      </dl><dl><dt>

          <code><a href="mk_subread_index.html">mk_subread_index()</a></code>

        </dt>
        <dd>Make a Subread index</dd>
      </dl><dl><dt>

          <code><a href="download_accessions.html">download_accessions()</a></code>

        </dt>
        <dd>Download indexes required for MetaScope ID and MetaBlast modules</dd>
      </dl></div><div class="section level2">
      <h2 id="metaalign">MetaAlign<a class="anchor" aria-label="anchor" href="#metaalign"></a></h2>

      <div class="section-desc"><p>Align sequencing reads to indexed target genome sequences</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="align_target.html">align_target()</a></code>

        </dt>
        <dd>Align microbiome reads to a set of reference libraries</dd>
      </dl><dl><dt>

          <code><a href="align_target_bowtie.html">align_target_bowtie()</a></code>

        </dt>
        <dd>Align microbiome reads to set of indexed Bowtie2 libraries</dd>
      </dl><dl><dt>

          <code><a href="align_details.html">align_details</a></code>

        </dt>
        <dd>A universal parameter settings object for Rsubread alignment</dd>
      </dl><dl><dt>

          <code><a href="bt2_16S_params.html">bt2_16S_params</a></code>

        </dt>
        <dd>A universal parameter object for Bowtie 2 16S alignment</dd>
      </dl><dl><dt>

          <code><a href="bt2_regular_params.html">bt2_regular_params</a></code>

        </dt>
        <dd>A universal parameter object for Bowtie 2 metagenomic or non-16S alignment</dd>
      </dl><dl><dt>

          <code><a href="bt2_missing_params.html">bt2_missing_params</a></code>

        </dt>
        <dd>A universal parameter object for Bowtie 2 metagenomic alignment where the host genome is thought to be absent from the reference database</dd>
      </dl></div><div class="section level2">
      <h2 id="metafilter">MetaFilter<a class="anchor" aria-label="anchor" href="#metafilter"></a></h2>

      <div class="section-desc"><p>Remove reads mapped to indexed host genome sequences</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="filter_host.html">filter_host()</a></code>

        </dt>
        <dd>Use Rsubread to align reads against one or more filter libraries and subsequently remove mapped reads</dd>
      </dl><dl><dt>

          <code><a href="filter_host_bowtie.html">filter_host_bowtie()</a></code>

        </dt>
        <dd>Use Rbowtie2 to align reads against one or more filter libraries and subsequently remove mapped reads</dd>
      </dl></div><div class="section level2">
      <h2 id="metaid">MetaID<a class="anchor" aria-label="anchor" href="#metaid"></a></h2>

      <div class="section-desc"><p>Reassign ambiguously mapped reads to probable source genome</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="metascope_id.html">metascope_id()</a></code>

        </dt>
        <dd>Identify which genomes are represented in a processed sample</dd>
      </dl></div><div class="section level2">
      <h2 id="metacombine">MetaCombine<a class="anchor" aria-label="anchor" href="#metacombine"></a></h2>

      <div class="section-desc"><p>Aggregate samples into a MultiAssayExperiment compatible with the animalcules package</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="convert_animalcules.html">convert_animalcules()</a></code>

        </dt>
        <dd>Create a multi-assay experiment from MetaScope output for usage with animalcules</dd>
      </dl><dl><dt>

          <code><a href="convert_animalcules_silva.html">convert_animalcules_silva()</a></code>

        </dt>
        <dd>Create a multi-assay experiment from MetaScope output for usage with animalcules with the SILVA 13_8 database</dd>
      </dl><dl><dt>

          <code><a href="convert_animalcules_patho.html">convert_animalcules_patho()</a></code>

        </dt>
        <dd>Create a multi-assay experiment from PathoScope 2.0 output for usage with animalcules</dd>
      </dl></div><div class="section level2">
      <h2 id="helper-functions">Helper Functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a></h2>

      <div class="section-desc"><p>These functions were developed to aid in carrying out other primary functions.</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="remove_matches.html">remove_matches()</a></code>

        </dt>
        <dd>Helper function to remove reads matched to filter libraries</dd>
      </dl><dl><dt>

          <code><a href="combined_header.html">combined_header()</a></code>

        </dt>
        <dd>Create a combined .bam header</dd>
      </dl><dl><dt>

          <code><a href="bam_reheader_R.html">bam_reheader_R()</a></code>

        </dt>
        <dd>Replace the header from a .bam file</dd>
      </dl><dl><dt>

          <code><a href="get_children.html">get_children()</a></code>

        </dt>
        <dd>Get child nodes from NCBI taxonomy</dd>
      </dl><dl><dt>

          <code><a href="merge_bam_files.html">merge_bam_files()</a></code>

        </dt>
        <dd>Merge multiple .bam files</dd>
      </dl><dl><dt>

          <code><a href="count_matches.html">count_matches()</a></code>

        </dt>
        <dd>Count the number of base lengths in a CIGAR string for a given operation</dd>
      </dl><dl><dt>

          <code><a href="extract_reads.html">extract_reads()</a></code>

        </dt>
        <dd>Helper function for demultiplexing</dd>
      </dl><dl><dt>

          <code><a href="locations.html">locations()</a></code>

        </dt>
        <dd>Helper Function for MetaScope ID</dd>
      </dl><dl><dt>

          <code><a href="filter_unmapped_reads.html">filter_unmapped_reads()</a></code>

        </dt>
        <dd>Filter unmapped reads</dd>
      </dl><dl><dt>

          <code><a href="check_samtools_exists.html">check_samtools_exists()</a></code>

        </dt>
        <dd>Check if samtools exists on the system</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

