<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Identify which genomes are represented in a processed sample — metascope_id • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Identify which genomes are represented in a processed sample — metascope_id"><meta name="description" content="This function will read in a .bam or .csv.gz file, annotate the taxonomy and
genome names, reduce the mapping ambiguity using a mixture model, and output
a .csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file)."><meta property="og:description" content="This function will read in a .bam or .csv.gz file, annotate the taxonomy and
genome names, reduce the mapping ambiguity using a mixture model, and output
a .csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file)."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.9.1</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/publicly_hosted_indices_and_accessions.html">Publicly Hosted Indices</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Package News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/MetaScope"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Identify which genomes are represented in a processed sample</h1>
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/R/metascope_id.R" class="external-link"><code>R/metascope_id.R</code></a></small>
      <div class="d-none name"><code>metascope_id.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function will read in a .bam or .csv.gz file, annotate the taxonomy and
genome names, reduce the mapping ambiguity using a mixture model, and output
a .csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">metascope_id</span><span class="op">(</span></span>
<span>  <span class="va">input_file</span>,</span>
<span>  input_type <span class="op">=</span> <span class="st">"csv.gz"</span>,</span>
<span>  aligner <span class="op">=</span> <span class="st">"bowtie2"</span>,</span>
<span>  db <span class="op">=</span> <span class="st">"ncbi"</span>,</span>
<span>  db_feature_table <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  accession_path <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  tmp_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">dirname</a></span><span class="op">(</span><span class="va">input_file</span><span class="op">)</span>,</span>
<span>  out_dir <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/basename.html" class="external-link">dirname</a></span><span class="op">(</span><span class="va">input_file</span><span class="op">)</span>,</span>
<span>  convEM <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">10000</span>,</span>
<span>  maxitsEM <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  update_bam <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  num_species_plot <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  out_fastas <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  num_genomes <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  num_reads <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-input-file">input_file<a class="anchor" aria-label="anchor" href="#arg-input-file"></a></dt>
<dd><p>The .bam or .csv.gz file of sample reads to be identified.</p></dd>


<dt id="arg-input-type">input_type<a class="anchor" aria-label="anchor" href="#arg-input-type"></a></dt>
<dd><p>Extension of file input. Should be either "bam" or
"csv.gz". Default is "csv.gz".</p></dd>


<dt id="arg-aligner">aligner<a class="anchor" aria-label="anchor" href="#arg-aligner"></a></dt>
<dd><p>The aligner which was used to create the bam file. Default is
"bowtie2" but can also be set to "subread" or "other".</p></dd>


<dt id="arg-db">db<a class="anchor" aria-label="anchor" href="#arg-db"></a></dt>
<dd><p>Currently accepts one of <code>c("ncbi", "silva", "other")</code>
Default is <code>"ncbi"</code>, appropriate for samples aligned against indices
compiled from NCBI whole genome databases. Alternatively, usage of an
alternate database (like Greengenes2) should be specified with
<code>"other"</code>.</p></dd>


<dt id="arg-db-feature-table">db_feature_table<a class="anchor" aria-label="anchor" href="#arg-db-feature-table"></a></dt>
<dd><p>If <code>db = "other"</code>, a data.frame must be supplied
with two columns, "Feature ID" matching the names of the alignment indices,
and a second <code>character</code> column supplying the taxon identifying information.</p></dd>


<dt id="arg-accession-path">accession_path<a class="anchor" aria-label="anchor" href="#arg-accession-path"></a></dt>
<dd><p>(character) Filepath to NCBI accessions SQL database. See
<code>taxonomzr::prepareDatabase()</code>.</p></dd>


<dt id="arg-tmp-dir">tmp_dir<a class="anchor" aria-label="anchor" href="#arg-tmp-dir"></a></dt>
<dd><p>Path to a directory to which bam and updated bam files can be saved.
Required.</p></dd>


<dt id="arg-out-dir">out_dir<a class="anchor" aria-label="anchor" href="#arg-out-dir"></a></dt>
<dd><p>The directory to which the .csv output file will be output.
Defaults to <code>dirname(input_file)</code>.</p></dd>


<dt id="arg-convem">convEM<a class="anchor" aria-label="anchor" href="#arg-convem"></a></dt>
<dd><p>The convergence parameter of the EM algorithm. Default set at
<code>1/10000</code>.</p></dd>


<dt id="arg-maxitsem">maxitsEM<a class="anchor" aria-label="anchor" href="#arg-maxitsem"></a></dt>
<dd><p>The maximum number of EM iterations, regardless of whether
the convEM is below the threshhold. Default set at <code>50</code>. If set at
<code>0</code>, the algorithm skips the EM step and summarizes the .bam file 'as
is'.</p></dd>


<dt id="arg-update-bam">update_bam<a class="anchor" aria-label="anchor" href="#arg-update-bam"></a></dt>
<dd><p>Whether to update BAM file with new read assignments.
Default is <code>FALSE</code>. If <code>TRUE</code>, requires <code>input_type = "bam"</code>
such that a BAM file is the input to the function.</p></dd>


<dt id="arg-num-species-plot">num_species_plot<a class="anchor" aria-label="anchor" href="#arg-num-species-plot"></a></dt>
<dd><p>The number of genome coverage plots to be saved.
Default is <code>NULL</code>, which saves coverage plots for the ten most highly
abundant species.</p></dd>


<dt id="arg-out-fastas">out_fastas<a class="anchor" aria-label="anchor" href="#arg-out-fastas"></a></dt>
<dd><p>Logical, whether or not to output fasta files of reads.
Default is <code>FALSE</code>.</p></dd>


<dt id="arg-num-genomes">num_genomes<a class="anchor" aria-label="anchor" href="#arg-num-genomes"></a></dt>
<dd><p>Number of genomes to output fasta files for
<code>out_fastas</code>. Default is <code>100</code>.</p></dd>


<dt id="arg-num-reads">num_reads<a class="anchor" aria-label="anchor" href="#arg-num-reads"></a></dt>
<dd><p>Number of reads per genome per fasta file for
<code>out_fastas</code>. Default is <code>50</code>.</p></dd>


<dt id="arg-quiet">quiet<a class="anchor" aria-label="anchor" href="#arg-quiet"></a></dt>
<dd><p>Turns off most messages. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>This function returns a .csv file with annotated read counts to
  genomes with mapped reads. The function itself returns the output .csv file
  name. Depending on the parameters specified, can also output an updated
  BAM file, and fasta files for additional analysis downstream.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Align reads to reference library and then apply metascope_id()</span></span></span>
<span class="r-in"><span><span class="co">## Assuming filtered bam files already exist</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create temporary directory</span></span></span>
<span class="r-in"><span><span class="va">file_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">file_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get temporary accessions database</span></span></span>
<span class="r-in"><span><span class="va">tmp_accession</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"example_accessions.sql"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### Subread aligned bam file</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create object with path to filtered subread csv.gz file</span></span></span>
<span class="r-in"><span><span class="va">filt_file</span> <span class="op">&lt;-</span> <span class="st">"subread_target.filtered.csv.gz"</span></span></span>
<span class="r-in"><span><span class="va">bamPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">filt_file</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="va">bamPath</span>, <span class="va">file_temp</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Run metascope id with the aligner option set to subread</span></span></span>
<span class="r-in"><span><span class="fu">metascope_id</span><span class="op">(</span>input_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">file_temp</span>, <span class="va">filt_file</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>             aligner <span class="op">=</span> <span class="st">"subread"</span>, num_species_plot <span class="op">=</span> <span class="fl">0</span>,</span></span>
<span class="r-in"><span>             input_type <span class="op">=</span> <span class="st">"csv.gz"</span>, accession_path <span class="op">=</span> <span class="va">tmp_accession</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TaxonomyID Genome                   read_count Proportion readsEM EMProportion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                         <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 273036     Bacteria;Bacillota;Baci…        102       0.85   102.         0.847</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 418127     Bacteria;Bacillota;Baci…         18       0.15    18.3        0.153</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### Bowtie 2 aligned .csv.gz file</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create object with path to filtered bowtie2 bam file</span></span></span>
<span class="r-in"><span><span class="va">bowtie_file</span> <span class="op">&lt;-</span> <span class="st">"bowtie_target.filtered.csv.gz"</span></span></span>
<span class="r-in"><span><span class="va">bamPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="va">bowtie_file</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="va">bamPath</span>, <span class="va">file_temp</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Run metascope id with the aligner option set to bowtie2</span></span></span>
<span class="r-in"><span><span class="fu">metascope_id</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">file_temp</span>, <span class="va">bowtie_file</span><span class="op">)</span>, aligner <span class="op">=</span> <span class="st">"bowtie2"</span>,</span></span>
<span class="r-in"><span>             num_species_plot <span class="op">=</span> <span class="fl">0</span>, input_type <span class="op">=</span> <span class="st">"csv.gz"</span>,</span></span>
<span class="r-in"><span>             accession_path <span class="op">=</span> <span class="va">tmp_accession</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TaxonomyID  Genome                  read_count Proportion readsEM EMProportion</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> NC_001498.1 unknown genome; access…          6          1       6            1</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Remove temporary directory</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">file_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

