<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><meta name="description" content="This function will read in a .bam or .rds file, annotate the taxonomy and
genome names, reduce the mapping ambiguity using a mixture model, and output
a .csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file)."><title>Identify which genomes are represented in a sample — metascope_id • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.0/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.0/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Identify which genomes are represented in a sample — metascope_id"><meta property="og:description" content="This function will read in a .bam or .rds file, annotate the taxonomy and
genome names, reduce the mapping ambiguity using a mixture model, and output
a .csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file)."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-dark"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.99.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../index.html">Home</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/MetaScope_vignette.html">Tutorial</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Function Reference</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Package News</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/compbiomed/MetaScope">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Identify which genomes are represented in a sample</h1>
      <small class="dont-index">Source: <a href="https://github.com/compbiomed/MetaScope/blob/HEAD/R/MetaScope_ID.R" class="external-link"><code>R/MetaScope_ID.R</code></a></small>
      <div class="d-none name"><code>metascope_id.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>This function will read in a .bam or .rds file, annotate the taxonomy and
genome names, reduce the mapping ambiguity using a mixture model, and output
a .csv file with the results. Currently, it assumes that the genome
library/.bam files use NCBI accession names for reference names (rnames in
.bam file).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">metascope_id</span><span class="op">(</span></span>
<span>  <span class="va">input_file</span>,</span>
<span>  input_type <span class="op">=</span> <span class="st">"csv.gz"</span>,</span>
<span>  aligner <span class="op">=</span> <span class="st">"bowtie2"</span>,</span>
<span>  NCBI_key <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  out_file <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="fu">tools</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/r/tools/fileutils.html" class="external-link">file_path_sans_ext</a></span><span class="op">(</span><span class="va">input_file</span><span class="op">)</span>, <span class="st">".metascope_id.csv"</span><span class="op">)</span>,</span>
<span>  convEM <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">10000</span>,</span>
<span>  maxitsEM <span class="op">=</span> <span class="fl">25</span>,</span>
<span>  num_species_plot <span class="op">=</span> <span class="cn">NULL</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>
    <dl><dt>input_file</dt>
<dd><p>The .bam or .rds file that needs to be identified.</p></dd>


<dt>input_type</dt>
<dd><p>Extension of file input. Should be either "bam" or
"csv.gz". Default is "csv.gz".</p></dd>


<dt>aligner</dt>
<dd><p>The aligner which was used to create the bam file. Default is
"bowtie2" but can also be set to "subread" or "other".</p></dd>


<dt>NCBI_key</dt>
<dd><p>(character) NCBI Entrez API key. optional. See
taxize::use_entrez(). Due to the high number of requests made to NCBI, this
function will be less prone to errors if you obtain an NCBI key. You may
enter the string as an input or set it as ENTREZ_KEY in .Renviron.</p></dd>


<dt>out_file</dt>
<dd><p>The name of the .csv output file. Defaults to the input_file
basename plus ".metascope_id.csv".</p></dd>


<dt>convEM</dt>
<dd><p>The convergence parameter of the EM algorithm. Default set at
<code>1/10000</code>.</p></dd>


<dt>maxitsEM</dt>
<dd><p>The maximum number of EM iterations, regardless of whether
the convEM is below the threshhold. Default set at <code>50</code>. If set at
<code>0</code>, the algorithm skips the EM step and summarizes the .bam file 'as
is'</p></dd>


<dt>num_species_plot</dt>
<dd><p>The number of genome coverage plots to be saved.
Default is <code>NULL</code>, which saves coverage plots for the ten most highly
abundant species.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    

<p>This function returns a .csv file with annotated read counts to
  genomes with mapped reads. The function itself returns the output .csv file
  name.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="co">#### Align reads to reference library and then apply metascope_id()</span></span></span>
<span class="r-in"><span><span class="co">## Assuming filtered bam files already exist</span></span></span>
<span class="r-in"><span><span class="va">file_temp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/tempfile.html" class="external-link">tempfile</a></span><span class="op">(</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files2.html" class="external-link">dir.create</a></span><span class="op">(</span><span class="va">file_temp</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### Subread aligned bam file</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Create object with path to filtered subread csv.gz file</span></span></span>
<span class="r-in"><span><span class="va">bamPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"subread_target.filtered.csv.gz"</span>,</span></span>
<span class="r-in"><span>                       package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">copied_csv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">file_temp</span>, <span class="st">"subread_target.filtered.csv.gz"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="va">bamPath</span>, <span class="va">copied_csv</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Run metascope id with the aligner option set to bowtie2</span></span></span>
<span class="r-in"><span><span class="fu">metascope_id</span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">copied_csv</span>, aligner <span class="op">=</span> <span class="st">"subread"</span>,</span></span>
<span class="r-in"><span>             num_species_plot <span class="op">=</span> <span class="fl">0</span>, input_type <span class="op">=</span> <span class="st">"csv.gz"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading .csv.gz file: /scratch/141165.1.ood/RtmpwygBTw/file7c3349f1029f/subread_target.filtered.csv.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 120 reads aligned to 15 NCBI accessions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Obtaining taxonomy and genome names</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No ENTREZ API key provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Get one via taxize::use_entrez()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 2 unique NCBI taxonomy IDs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting up the EM algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting EM iterations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 1100.527318326282</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20.00694441476995597</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 30.00032363784016558</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 41.49008395637793e-05</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	DONE! Converged in 4 iterations.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found reads for 2 genomes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Results written to /scratch/141165.1.ood/RtmpwygBTw/file7c3349f1029f/subread_target.filtered.csv.metascope_id.csv</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No coverage plots created</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 2 × 6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TaxonomyID Genome                              read_…¹ Propo…² readsEM EMPro…³</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 273036     Staphylococcus aureus RF122, compl…     102    0.85   102.    0.847</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">2</span> 418127     Staphylococcus aureus subsp. aureu…      18    0.15    18.3   0.153</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with abbreviated variable names ¹​read_count, ²​Proportion, ³​EMProportion</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">#### Bowtie 2 aligned .csv.gz file</span></span></span>
<span class="r-in"><span><span class="co">## Create object with path to filtered bowtie2 bam file</span></span></span>
<span class="r-in"><span><span class="va">bamPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,<span class="st">"bowtie_target.filtered.csv.gz"</span>,</span></span>
<span class="r-in"><span>                       package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">copied_csv</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/file.path.html" class="external-link">file.path</a></span><span class="op">(</span><span class="va">file_temp</span>, <span class="st">"bowtie_target.filtered.csv.gz"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/files.html" class="external-link">file.copy</a></span><span class="op">(</span><span class="va">bamPath</span>, <span class="va">copied_csv</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] TRUE</span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Run metascope id with the aligner option set to bowtie2</span></span></span>
<span class="r-in"><span><span class="fu">metascope_id</span><span class="op">(</span>input_file <span class="op">=</span> <span class="va">copied_csv</span>, aligner <span class="op">=</span> <span class="st">"bowtie2"</span>,</span></span>
<span class="r-in"><span>             num_species_plot <span class="op">=</span> <span class="fl">0</span>, input_type <span class="op">=</span> <span class="st">"csv.gz"</span><span class="op">)</span></span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Reading .csv.gz file: /scratch/141165.1.ood/RtmpwygBTw/file7c3349f1029f/bowtie_target.filtered.csv.gz</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 6 reads aligned to 1 NCBI accessions</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Obtaining taxonomy and genome names</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No ENTREZ API key provided</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span>  Get one via taxize::use_entrez()</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> See https://ncbiinsights.ncbi.nlm.nih.gov/2017/11/02/new-api-keys-for-the-e-utilities/</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	Found 1 unique NCBI taxonomy IDs</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Setting up the EM algorithm</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Starting EM iterations</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 15</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 20</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> 	DONE! Converged in 2 iterations.</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Found reads for 1 genomes</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> Results written to /scratch/141165.1.ood/RtmpwygBTw/file7c3349f1029f/bowtie_target.filtered.csv.metascope_id.csv</span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> No coverage plots created</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># A tibble: 1 × 6</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   TaxonomyID Genome                         read_count Proport…¹ readsEM EMPro…²</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #BCBCBC;">1</span> 11234      Measles virus, complete genome          6         1       6       1</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> <span style="color: #949494;"># … with abbreviated variable names ¹​Proportion, ²​EMProportion</span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/unlink.html" class="external-link">unlink</a></span><span class="op">(</span><span class="va">file_temp</span>, recursive <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Site created by Aubrey Odom-Mabey.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.5.</p>
</div>

    </footer></div>

  

  

  </body></html>

