<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Demultiplexing sequencing reads — meta_demultiplex • MetaScope</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Demultiplexing sequencing reads — meta_demultiplex"><meta name="description" content="Function for demultiplexing sequencing reads arranged in a common format
provided by sequencers (such as Illumina) generally for 16S data. This
function takes a matrix of sample names/barcodes, a .fastq file of barcodes
by sequence header, and a .fastq file of reads corresponding to the barcodes.
Based on the barcodes given, the function extracts all reads for the indexed
barcode and writes all the reads from that barcode to separate .fastq files."><meta property="og:description" content="Function for demultiplexing sequencing reads arranged in a common format
provided by sequencers (such as Illumina) generally for 16S data. This
function takes a matrix of sample names/barcodes, a .fastq file of barcodes
by sequence header, and a .fastq file of reads corresponding to the barcodes.
Based on the barcodes given, the function extracts all reads for the indexed
barcode and writes all the reads from that barcode to separate .fastq files."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-dark" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">MetaScope</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.6</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">Home</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-tutorials" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Tutorials</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-tutorials"><li><a class="dropdown-item" href="../articles/MetaScope_vignette.html">Introduction to MetaScope</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/publicly_hosted_indices_and_accessions.html">Publicly Hosted Indices</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/SILVA.html">Usage with SILVA 16S Database</a></li>
    <li><a class="dropdown-item" href="../articles/vignette_docs/Example_Script.html">Example MetaScope Script</a></li>
  </ul></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Function Reference</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Package News</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><a class="external-link nav-link" href="https://github.com/wejlab/MetaScope"><span class="fa fa-github"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Demultiplexing sequencing reads</h1>
      <small class="dont-index">Source: <a href="https://github.com/wejlab/MetaScope/blob/HEAD/R/demultiplex.R" class="external-link"><code>R/demultiplex.R</code></a></small>
      <div class="d-none name"><code>meta_demultiplex.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Function for demultiplexing sequencing reads arranged in a common format
provided by sequencers (such as Illumina) generally for 16S data. This
function takes a matrix of sample names/barcodes, a .fastq file of barcodes
by sequence header, and a .fastq file of reads corresponding to the barcodes.
Based on the barcodes given, the function extracts all reads for the indexed
barcode and writes all the reads from that barcode to separate .fastq files.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">meta_demultiplex</span><span class="op">(</span></span>
<span>  <span class="va">barcodeFile</span>,</span>
<span>  <span class="va">indexFile</span>,</span>
<span>  <span class="va">readFile</span>,</span>
<span>  rcBarcodes <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  location <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  threads <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  hammingDist <span class="op">=</span> <span class="fl">0</span>,</span>
<span>  quiet <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-barcodefile">barcodeFile<a class="anchor" aria-label="anchor" href="#arg-barcodefile"></a></dt>
<dd><p>Path to a file containing a .tsv matrix with a
header row, and then sample names (column 1) and barcodes (column 2).</p></dd>


<dt id="arg-indexfile">indexFile<a class="anchor" aria-label="anchor" href="#arg-indexfile"></a></dt>
<dd><p>Path to a .fastq file that contains the barcodes for
each read. The headers should be the same (and in the same order) as
<code>readFile</code>, and the sequence in the <code>indexFile</code> should be the
corresponding barcode for each read. Quality scores are not considered.</p></dd>


<dt id="arg-readfile">readFile<a class="anchor" aria-label="anchor" href="#arg-readfile"></a></dt>
<dd><p>Path to the sequencing read .fastq file that corresponds
to the <code>indexFile</code>.</p></dd>


<dt id="arg-rcbarcodes">rcBarcodes<a class="anchor" aria-label="anchor" href="#arg-rcbarcodes"></a></dt>
<dd><p>Should the barcode indexes in the barcodeFile be reverse
complemented to match the sequences in the <code>indexFile</code>? Defaults to
<code>TRUE</code>.</p></dd>


<dt id="arg-location">location<a class="anchor" aria-label="anchor" href="#arg-location"></a></dt>
<dd><p>A directory location to store the demultiplexed read files.
Defaults to generate a new temporary directory.</p></dd>


<dt id="arg-threads">threads<a class="anchor" aria-label="anchor" href="#arg-threads"></a></dt>
<dd><p>The number of threads to use for parallelization
(BiocParallel). This function will parallelize over the barcodes and
extract reads for each barcode separately and write them to separate
demultiplexed files.</p></dd>


<dt id="arg-hammingdist">hammingDist<a class="anchor" aria-label="anchor" href="#arg-hammingdist"></a></dt>
<dd><p>Uses a Hamming Distance or number of base differences to
allow for inexact matches for the barcodes/indexes. Defaults to <code>0</code>.
Warning: if the Hamming Distance is <code>&gt;=1</code> and this leads to inexact
index matches to more than one barcode, that read will be written to more
than one demultiplexed read files.</p></dd>


<dt id="arg-quiet">quiet<a class="anchor" aria-label="anchor" href="#arg-quiet"></a></dt>
<dd><p>Turns off most messages. Default is <code>TRUE</code>.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>Returns multiple .fastq files that contain all reads whose index
  matches the barcodes given. These files will be written to the location
  directory, and will be named based on the given sampleNames and barcodes,
  e.g. './demultiplex_fastq/SampleName1_GGAATTATCGGT.fastq.gz'</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Get barcode, index, and read data locations</span></span></span>
<span class="r-in"><span><span class="va">barcodePath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"barcodes.txt"</span>, package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">indexPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example_index.fastq"</span>,</span></span>
<span class="r-in"><span>                         package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">readPath</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"virus_example.fastq"</span>,</span></span>
<span class="r-in"><span>                         package <span class="op">=</span> <span class="st">"MetaScope"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Demultiplex</span></span></span>
<span class="r-in"><span><span class="va">demult</span> <span class="op">&lt;-</span> <span class="fu">meta_demultiplex</span><span class="op">(</span><span class="va">barcodePath</span>, <span class="va">indexPath</span>, <span class="va">readPath</span>, rcBarcodes <span class="op">=</span> <span class="cn">FALSE</span>,</span></span>
<span class="r-in"><span>                      hammingDist <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">demult</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   SampleName Barcode NumberOfReads</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 1        CDV TCCACGT            25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 2   LaCrosse ACAGGCT            25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 3        RSV ATCGTGC            25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 4       EboV ACTACAG            25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 5    Measles AAGTCGC            25</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> 6        VSV TCTCAGG            25</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Site created by Aubrey Odom.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

